diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java b/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
index a743aa3..326f0c3 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
@@ -327,7 +327,7 @@ public static BgReading create(double raw_data, double filtered_data, Context co
                     bgReading.calculated_value = 9;
                     bgReading.hide_slope = true;
                 } else {
-                    bgReading.calculated_value = Math.min(400, Math.max(40, bgReading.calculated_value));
+                    bgReading.calculated_value = Math.min(400, Math.max(39, bgReading.calculated_value));
                 }
                 Log.w(TAG, "NEW VALUE CALCULATED AT: " + bgReading.calculated_value);
 
