diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
index f1e1fbb..98b0b48 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
@@ -371,8 +371,10 @@ public void run() {
                 NotificationManagerCompat
                         .from(mContext)
                         .notify(ongoingNotificationId, createOngoingNotification(bgGraphBuilder, mContext));
-                iconBitmap.recycle();
-                notifiationBitmap.recycle();
+                if (iconBitmap != null)
+                    iconBitmap.recycle();
+                if (notifiationBitmap != null)
+                    notifiationBitmap.recycle();
             }
         });
     }
