diff --git a/app/build.gradle b/app/build.gradle
index dadfd85..5c146ee 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -18,6 +18,7 @@ repositories {
     maven {
         url "https://jitpack.io"
     }
+    maven { url "https://oss.sonatype.org/content/groups/public/" }
 }
 
 def generateVersionName = { ->
@@ -98,7 +99,7 @@ dependencies {
     compile 'com.google.code.gson:gson:2.3'
     compile 'org.mongodb:mongo-java-driver:2.10.1'
     compile 'com.squareup.retrofit:retrofit:1.9.0'
-    compile 'com.getpebble:pebblekit:2.6.0@aar'
+    compile 'com.getpebble:pebblekit:3.0.0'
     compile 'io.reactivex:rxjava:1.0.0'
     compile 'ch.acra:acra:4.5.0'
 
