diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Models/TransmitterData.java b/app/src/main/java/com/eveningoutpost/dexdrip/Models/TransmitterData.java
index 5d44bb9..189ad46 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Models/TransmitterData.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Models/TransmitterData.java
@@ -47,7 +47,8 @@ public static TransmitterData create(byte[] buffer, int len, Long timestamp) {
             txData.put(buffer, 0, len);
             transmitterData.raw_data = txData.getInt(2);
             transmitterData.filtered_data = txData.getInt(6);
-            transmitterData.sensor_battery_level = txData.get(10);
+            //  bitwise and with 0xff (1111....1) to avoid that the byte is treated as signed.
+            transmitterData.sensor_battery_level = txData.get(10) & 0xff;
             Log.w(TAG, "Created transmitterData record with Raw value of " + transmitterData.raw_data + " and Filtered value of " + transmitterData.filtered_data + " at " + transmitterData.timestamp);
         } else { //this is NOT a dexbridge packet.  Process accordingly.
             Log.w(TAG, "create Processing a BTWixel or IPWixel packet");
