diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java b/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
index 8b567f8..06f990f 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
@@ -16,6 +16,7 @@
 import android.os.Bundle;
 import android.os.Handler;
 import android.preference.PreferenceManager;
+import android.text.InputFilter;
 import android.text.TextUtils;
 import android.util.Log;
 import android.view.LayoutInflater;
@@ -436,8 +437,10 @@ private void requestTransmitterId(final SharedPreferences prefs) {
         dialog.setContentView(R.layout.dialog_single_text_field);
         Button saveButton = (Button) dialog.findViewById(R.id.saveButton);
         Button cancelButton = (Button) dialog.findViewById(R.id.cancelButton);
+        dialog.findViewById(R.id.scannerButton).setVisibility(View.GONE);
         final EditText serialNumberView = (EditText) dialog.findViewById(R.id.editTextField);
         serialNumberView.setHint("00000");
+        serialNumberView.setFilters(new InputFilter[]{new InputFilter.AllCaps()});
         ((TextView) dialog.findViewById(R.id.instructionsTextField)).setText("Enter Your Dexcom Transmitter ID");
 
         saveButton.setOnClickListener(new View.OnClickListener() {
