diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java b/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
index e17f39c..b03014f 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
@@ -85,6 +85,7 @@ protected void onPause() {
         super.onPause();
         scanLeDevice(false);
         mLeDeviceListAdapter.clear();
+        mLeDeviceListAdapter.notifyDataSetChanged();
     }
 
     @Override
@@ -202,6 +203,7 @@ public LeDeviceListAdapter() {
         public void addDevice(BluetoothDevice device) {
             if(!mLeDevices.contains(device)) {
                 mLeDevices.add(device);
+                notifyDataSetChanged();
             }
         }
 
@@ -211,6 +213,7 @@ public BluetoothDevice getDevice(int position) {
 
         public void clear() {
             mLeDevices.clear();
+            notifyDataSetChanged();
         }
 
         @Override
@@ -249,7 +252,6 @@ public View getView(int i, View view, ViewGroup viewGroup) {
             else
                 viewHolder.deviceName.setText(R.string.unknown_device);
             viewHolder.deviceAddress.setText(device.getAddress());
-
             return view;
         }
     }
@@ -266,6 +268,7 @@ public void onLeScan(final BluetoothDevice device, int rssi, byte[] scanRecord)
                         public void run() {
                             mLeDeviceListAdapter.addDevice(device);
                             mLeDeviceListAdapter.notifyDataSetChanged();
+
                         }
                     });
                 }
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java b/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
index 3bedf0b..3f547be 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
@@ -151,9 +151,12 @@ public String displayValue(Context context) {
 
     public static double activeSlope() {
         BgReading bgReading = BgReading.lastNoSenssor();
-        double slope = (2 * bgReading.a * (new Date().getTime() + BESTOFFSET)) + bgReading.b;
-        Log.w(TAG, "ESTIMATE SLOPE" + slope);
-        return slope;
+        if (bgReading != null) {
+            double slope = (2 * bgReading.a * (new Date().getTime() + BESTOFFSET)) + bgReading.b;
+            Log.w(TAG, "ESTIMATE SLOPE" + slope);
+            return slope;
+        }
+        return 0;
     }
 
     public static double activePrediction() {
