diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java
index 1d425a9..918cce7 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java
@@ -11,6 +11,8 @@
 import android.os.PowerManager;
 import android.preference.PreferenceManager;
 import android.provider.BaseColumns;
+
+import com.eveningoutpost.dexdrip.Models.Calibration;
 import com.eveningoutpost.dexdrip.Models.UserError.Log;
 
 import com.activeandroid.Model;
@@ -111,6 +113,33 @@ public static void addToQueue(BgReading bgReading, String operation_type, Contex
                 bundle.putInt(Intents.EXTRA_SENSOR_BATTERY, getBatteryLevel(context));
                 bundle.putLong(Intents.EXTRA_TIMESTAMP, bgReading.timestamp);
 
+                //raw value
+                double slope = 0, intercept = 0, scale = 0, filtered = 0, unfiltered = 0, raw = 0;
+                Calibration cal = Calibration.last();
+                if (cal != null){;
+                    // slope/intercept/scale like uploaded to NightScout (NightScoutUploader.java)
+                    if(cal.check_in) {
+                        slope = cal.first_slope;
+                        intercept= cal.first_intercept;
+                        scale =  cal.first_scale;
+                    } else {
+                        slope = cal.slope * 1000;
+                        intercept=  (cal.intercept * -1000) / (cal.slope * 1000);
+                        scale = 1;
+                    }
+                    unfiltered= bgReading.usedRaw();
+                    filtered = bgReading.ageAdjustedFiltered();
+                }
+                //raw logic from https://github.com/nightscout/cgm-remote-monitor/blob/master/lib/plugins/rawbg.js#L59
+                if (slope != 0 && intercept != 0 && scale != 0) {
+                    if (filtered == 0 || bgReading.calculated_value < 40) {
+                        raw = scale * (unfiltered - intercept) / slope;
+                    } else {
+                        double ratio = scale * (filtered - intercept) / slope / bgReading.calculated_value;
+                        raw = scale * (unfiltered - intercept) / slope / ratio;
+                    }
+                }
+                bundle.putDouble(Intents.EXTRA_RAW, raw);
                 Intent intent = new Intent(Intents.ACTION_NEW_BG_ESTIMATE);
                 intent.putExtras(bundle);
                 intent.addFlags(Intent.FLAG_INCLUDE_STOPPED_PACKAGES);
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Intents.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Intents.java
index 75cdbd9..01a1ca1 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Intents.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Intents.java
@@ -12,6 +12,8 @@
     String EXTRA_BG_SLOPE_NAME = "com.eveningoutpost.dexdrip.Extras.BgSlopeName";
     String EXTRA_SENSOR_BATTERY = "com.eveningoutpost.dexdrip.Extras.SensorBattery";
     String EXTRA_TIMESTAMP = "com.eveningoutpost.dexdrip.Extras.Time";
+    String EXTRA_RAW = "com.eveningoutpost.dexdrip.Extras.Raw";
+
 
     String ACTION_NEW_BG_ESTIMATE_NO_DATA = "com.eveningoutpost.dexdrip.BgEstimateNoData";
 }
