diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
index 0e10fd9..0ee5280 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
@@ -164,6 +164,16 @@ public void FileBasedNotifications(Context context) {
 
             if (activeBgAlert.uuid.equals(newAlert.uuid)) {
                 // This is the same alert. Might need to play again...
+
+                //disable alert on stale data
+                if(prefs.getBoolean("disable_alerts_stale_data", false)) {
+                    int minutes = Integer.parseInt(prefs.getString("disable_alerts_stale_data_minutes", "15")) + 2;
+                    if ((new Date().getTime()) - (60000 * minutes) - BgReading.lastNoSenssor().timestamp > 0) {
+                        Log.d(TAG, "FileBasedNotifications : active alert found but not replaying it because more than three readings missed :  " + newAlert.name);
+                        return;
+                    }
+                }
+
                 Log.d(TAG, "FileBasedNotifications we have found an active alert, checking if we need to play it " + newAlert.name);
                 boolean trendingToAlertEnd = trendingToAlertEnd(context, false, newAlert);
                 AlertPlayer.getPlayer().ClockTick(context, trendingToAlertEnd, EditAlertActivity.unitsConvert2Disp(doMgdl, bgReading.calculated_value));
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java b/app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java
index 56a3d52..4461df5 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java
@@ -21,6 +21,8 @@
 import android.preference.RingtonePreference;
 import android.text.InputFilter;
 import android.text.TextUtils;
+import android.widget.Toast;
+
 import com.eveningoutpost.dexdrip.Models.UserError.Log;
 
 import com.eveningoutpost.dexdrip.R;
@@ -245,6 +247,7 @@ public void onCreate(Bundle savedInstanceState) {
             bindPreferenceSummaryToValueAndEnsureNumeric(findPreference("calibration_snooze"));
             bindPreferenceSummaryToValueAndEnsureNumeric(findPreference("bg_unclear_readings_minutes"));
             bindPreferenceSummaryToValueAndEnsureNumeric(findPreference("bg_missed_minutes"));
+            bindPreferenceSummaryToValueAndEnsureNumeric(findPreference("disable_alerts_stale_data_minutes"));
             bindPreferenceSummaryToValue(findPreference("falling_bg_val"));
             bindPreferenceSummaryToValue(findPreference("rising_bg_val"));
             bindPreferenceSummaryToValue(findPreference("other_alerts_sound"));
@@ -279,6 +282,22 @@ public void onCreate(Bundle savedInstanceState) {
             final PreferenceScreen calibrationAlertsScreen = (PreferenceScreen) findPreference("calibration_alerts_screen");
             final PreferenceCategory alertsCategory = (PreferenceCategory) findPreference("alerts_category");
             final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getActivity());
+            final Preference disableAlertsStaleDataMinutes = findPreference("disable_alerts_stale_data_minutes");
+            disableAlertsStaleDataMinutes.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+                @Override
+                public boolean onPreferenceChange(Preference preference, Object newValue) {
+                    if (!isNumeric(newValue.toString())) {
+                        return false;
+                    }
+                    if ((Integer.parseInt(newValue.toString())) < 10 ) {
+                        Toast.makeText(preference.getContext(),
+                                "Value must be at least 10 minutes", Toast.LENGTH_LONG).show();
+                        return false;
+                    }
+                    preference.setSummary(newValue.toString());
+                    return true;
+                }
+            });
             Log.d(TAG, prefs.getString("dex_collection_method", "BluetoothWixel"));
             if(prefs.getString("dex_collection_method", "BluetoothWixel").compareTo("DexcomShare") != 0) {
                 collectionCategory.removePreference(shareKey);
diff --git a/app/src/main/res/xml/pref_notifications.xml b/app/src/main/res/xml/pref_notifications.xml
index 3cc73d8..7970a21 100644
--- a/app/src/main/res/xml/pref_notifications.xml
+++ b/app/src/main/res/xml/pref_notifications.xml
@@ -38,6 +38,20 @@
                 android:title="Shortcut to Bg Level Alerts"
                 android:summary="Create a shortcut from main navigation to bg level screen"
                 android:defaultValue="false" />
+
+                <CheckBoxPreference
+                    android:key="disable_alerts_stale_data"
+                    android:title="Suppress Alerts if missed readings"
+                    android:summary="Suppress snoozed and active alerts after predefined period of missed readings"
+                    android:defaultValue="false" />
+                <EditTextPreference
+                    android:dependency="disable_alerts_stale_data"
+                    android:key="disable_alerts_stale_data_minutes"
+                    android:title="Suppress snoozed and active alerts after .. minutes (minimum 10)"
+                    android:inputType="number"
+                    android:summary="Suppress snoozed and active alerts after .. minutes (minimum 10)"
+                    android:defaultValue="15" />
+
         </PreferenceScreen>
 
 
@@ -103,7 +117,7 @@
                     android:summary="Alert if no data receieved in x minutes"
                     android:defaultValue="30" />
             </PreferenceCategory>
-
+            
             <PreferenceCategory
                 android:title="Falling/Rising BG">
                 <CheckBoxPreference
