diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Services/MissedReadingService.java b/app/src/main/java/com/eveningoutpost/dexdrip/Services/MissedReadingService.java
index 90a9bc1..2b1410f 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Services/MissedReadingService.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Services/MissedReadingService.java
@@ -22,6 +22,7 @@
 import com.eveningoutpost.dexdrip.UtilityModels.Notifications;
 
 import java.util.Calendar;
+import java.util.Date;
 
 public class MissedReadingService extends IntentService {
     SharedPreferences prefs;
@@ -42,20 +43,26 @@ protected void onHandleIntent(Intent intent) {
         bg_missed_minutes =  Integer.parseInt(prefs.getString("bg_missed_minutes", "30"));
         otherAlertSnooze =  Integer.parseInt(prefs.getString("other_alerts_snooze", "20"));
 
-        if (bg_missed_alerts && BgReading.getTimeSinceLastReading() > (bg_missed_minutes * 1000 * 60)) {
+        if (bg_missed_alerts && BgReading.getTimeSinceLastReading() > (bg_missed_minutes * 1000 * 60)
+                &&
+                prefs.getLong("alerts_disabled_until", 0) < new Date().getTime()) {
             Notifications.bgMissedAlert(mContext);
             checkBackAfterSnoozeTime();
-        } else {
-            checkBackAfterMissedTime();
+        } else  {
+            long alarmIn = prefs.getLong("alerts_disabled_until", 0) - new Date().getTime();
+            if (alarmIn <= 0) {
+                alarmIn = Long.parseLong(prefs.getString("bg_missed_minutes", "30"))* 1000 * 60;
+            }
+            checkBackAfterMissedTime(alarmIn);
         }
     }
 
-   public void checkBackAfterSnoozeTime() {
-       setAlarm(otherAlertSnooze * 1000 * 60);
-   }
+    public void checkBackAfterSnoozeTime() {
+        setAlarm(otherAlertSnooze * 1000 * 60);
+    }
 
-    public void checkBackAfterMissedTime() {
-        setAlarm(bg_missed_minutes * 1000 * 60);
+    public void checkBackAfterMissedTime(long alarmIn) {
+        setAlarm(alarmIn);
     }
 
     public void setAlarm(long alarmIn) {
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java b/app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java
index 8275e55..b48b276 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java
@@ -28,6 +28,7 @@
 
 import com.eveningoutpost.dexdrip.Models.ActiveBgAlert;
 import com.eveningoutpost.dexdrip.Models.AlertType;
+import com.eveningoutpost.dexdrip.Services.MissedReadingService;
 import com.eveningoutpost.dexdrip.UtilityModels.AlertPlayer;
 import com.eveningoutpost.dexdrip.UtilityModels.BgGraphBuilder;
 import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
@@ -174,6 +175,11 @@ private void setOnClickListenerOnClearDisabledButton(Button button, String alert
         button.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
                 prefs.edit().putLong(theAlert, 0).apply();
+                if (theAlert.equalsIgnoreCase("alerts_disabled_until")) {
+                    //this is needed to make sure that the missedreading alert will be rechecked, it might have to be raised
+                    //and if not (ie no missed readings for long enough) then the alarm should be reset because it might have to recheck the missedreading status sooner
+                    getApplicationContext().startService(new Intent(getApplicationContext(), MissedReadingService.class));
+                }
                 showDisableEnableButtons();
             }
         });
