diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Home.java b/app/src/main/java/com/eveningoutpost/dexdrip/Home.java
index a60aa2c..600206c 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Home.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Home.java
@@ -210,6 +210,7 @@ public void setColorFilter(ColorFilter cf) {}
         previewChart.setZoomType(ZoomType.HORIZONTAL);
 
         chart.setLineChartData(bgGraphBuilder.lineData());
+        chart.setOnValueTouchListener(bgGraphBuilder.getOnValueSelectTooltipListener());
         previewChart.setLineChartData(bgGraphBuilder.previewLineData());
         updateStuff = true;
 
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
index 6d2e450..2196890 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
@@ -6,6 +6,7 @@
 import android.graphics.Color;
 import android.preference.PreferenceManager;
 import android.text.format.DateFormat;
+import android.widget.Toast;
 
 import com.eveningoutpost.dexdrip.Models.BgReading;
 import com.eveningoutpost.dexdrip.Models.Calibration;
@@ -21,6 +22,7 @@
 import java.util.Locale;
 import java.util.TimeZone;
 
+import lecho.lib.hellocharts.listener.LineChartOnValueSelectListener;
 import lecho.lib.hellocharts.model.Axis;
 import lecho.lib.hellocharts.model.AxisValue;
 import lecho.lib.hellocharts.model.Line;
@@ -396,6 +398,7 @@ public String unitizedDeltaString(boolean showUnit, boolean highGranularity) {
         }
     }
 
+
     public static double mmolConvert(double mgdl) {
         return mgdl * Constants.MGDL_TO_MMOLL;
     }
@@ -408,4 +411,30 @@ public String unit() {
         }
 
     }
+
+    public OnValueSelectTooltipListener getOnValueSelectTooltipListener(){
+        return new OnValueSelectTooltipListener();
+    }
+
+    public class OnValueSelectTooltipListener implements LineChartOnValueSelectListener{
+
+        private Toast tooltip;
+
+        @Override
+        public synchronized void onValueSelected(int i, int i1, PointValue pointValue) {
+            final java.text.DateFormat timeFormat = DateFormat.getTimeFormat(context);
+            //Won't give the exact time of the reading but the time on the grid: close enough.
+            Long time = ((long)pointValue.getX())*FUZZER;
+            if(tooltip!= null){
+                tooltip.cancel();
+            }
+            tooltip = Toast.makeText(context, timeFormat.format(time)+ ": " + Math.round(pointValue.getY()*10)/ 10d , Toast.LENGTH_LONG);
+            tooltip.show();
+        }
+
+        @Override
+        public void onValueDeselected() {
+            // do nothing
+        }
+    }
 }
