diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Models/CalibrationRequest.java b/app/src/main/java/com/eveningoutpost/dexdrip/Models/CalibrationRequest.java
index 7e3c7f0..1a8d3ef 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Models/CalibrationRequest.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Models/CalibrationRequest.java
@@ -36,6 +36,7 @@ public static void createOffset(double center, double distance) {
         calibrationRequest.requestIfBelow = max;
         calibrationRequest.save();
 
+        calibrationRequest = new CalibrationRequest();
         calibrationRequest.requestIfAbove = min;
         calibrationRequest.requestIfBelow = center - distance;
         calibrationRequest.save();
@@ -58,10 +59,6 @@ public static boolean shouldRequestCalibration(BgReading bgReading){
                 .where("requestIfAbove < ?", bgReading.calculated_value)
                 .where("requestIfBelow > ?", bgReading.calculated_value)
                 .executeSingle();
-        if (calibrationRequest != null && Math.abs(bgReading.calculated_value_slope * 60000) < 1.8) {
-            return true;
-        } else {
-            return false;
-        }
+        return (calibrationRequest != null && Math.abs(bgReading.calculated_value_slope * 60000) < 1);
     }
 }
