diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/utils/BgToSpeech.java b/app/src/main/java/com/eveningoutpost/dexdrip/utils/BgToSpeech.java
index 4b7e552..91d9d2b 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/utils/BgToSpeech.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/utils/BgToSpeech.java
@@ -53,8 +53,10 @@ public static void speak(final double value, long timestamp){
     }
 
     private void tearDown() {
-        tts.shutdown();
-        tts = null;
+        if (tts != null) {
+            tts.shutdown();
+            tts = null;
+        }
     }
 
     private BgToSpeech(Context context){
