diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java b/app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java
index 9e8ef43..5103b29 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java
@@ -36,7 +36,9 @@
     private static BgToSpeech bgToSpeech;
 
     private final Context mContext;
-    PowerManager.WakeLock wakeLock; 
+    PowerManager.WakeLock wakeLock;
+
+    private final static long DEXCOM_PERIOD=300000;
     
     private static int lockCounter = 0;
     
@@ -301,7 +303,7 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
     }
 
     static Long timeForNextRead() {
-        final long DEXCOM_PERIOD=300000;
+
         TransmitterData lastTransmitterData = TransmitterData.last();
         if(lastTransmitterData == null) {
             // We did not receive a packet, well someone hopefully is looking at data, return relatively fast
@@ -353,9 +355,17 @@ public void readData()
     	TransmitterData lastTransmitterData = TransmitterData.last();
     	if(lastTransmitterData != null) {
     	    LastReportedTime = lastTransmitterData.timestamp;
-    	}
+
+            // jamorham fix to avoid going twice to network when we just got a packet
+            if ((new Date().getTime() - LastReportedTime) < DEXCOM_PERIOD-2000) {
+            Log.d(TAG, "Already have a recent packet - returning");
+            return;
+            }
+
+
+        }
     	Long startReadTime = LastReportedTime;
-    	
+
     	TransmitterRawData LastReportedReading = null;
     	Log.d(TAG, "Starting... LastReportedReading " + LastReportedReading);
     	// try to read one object...
