diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java b/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
index 64ea052..5df5b6b 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
@@ -318,8 +318,12 @@ protected void onListItemClick(ListView l, View v, int position, long id) {
             } else returnToHome();
 
         } else if(device.getName().toLowerCase().contains("drip")) {
-            if (!CollectionServiceStarter.isBTWixel(getApplicationContext()))
+            if (!
+                    (CollectionServiceStarter.isBTWixel(getApplicationContext())
+                            || CollectionServiceStarter.isWifiandBTWixel(getApplicationContext())
+                    )) {
                 prefs.edit().putString("dex_collection_method", "BluetoothWixel").apply();
+            }
             returnToHome();
         } else {
             returnToHome();
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java b/app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java
index d44dfe8..8733f44 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java
@@ -84,7 +84,7 @@ public NavDrawerBuilder(Context aContext) {
         }
 
         if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR2) {
-            if(CollectionServiceStarter.isBTWixel(context) || CollectionServiceStarter.isDexbridgeWixel(context)|| CollectionServiceStarter.isBTShare(context)) {
+            if(CollectionServiceStarter.isBTWixel(context) || CollectionServiceStarter.isDexbridgeWixel(context)|| CollectionServiceStarter.isBTShare(context) ||CollectionServiceStarter.isWifiandBTWixel(context)) {
                 this.nav_drawer_options.add(BluetoothScan.menu_name);
                 this.nav_drawer_intents.add(new Intent(context, BluetoothScan.class));
             }
