diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java b/app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java
index de115ec..1491545 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java
@@ -1,8 +1,10 @@
 package com.eveningoutpost.dexdrip;
 
 import android.app.Application;
+import android.database.sqlite.SQLiteDatabase;
 import android.preference.PreferenceManager;
 
+import com.activeandroid.Cache;
 import com.crashlytics.android.Crashlytics;
 import com.eveningoutpost.dexdrip.UtilityModels.CollectionServiceStarter;
 import com.eveningoutpost.dexdrip.UtilityModels.IdempotentMigrations;
@@ -19,12 +21,16 @@
     public void onCreate() {
         super.onCreate();
         Fabric.with(this, new Crashlytics());
-        CollectionServiceStarter collectionServiceStarter = new CollectionServiceStarter(getApplicationContext());
-        collectionServiceStarter.start(getApplicationContext());
+        CollectionServiceStarter collectionServiceStarter = new CollectionServiceStarter(this);
+        collectionServiceStarter.start(this);
         PreferenceManager.setDefaultValues(this, R.xml.pref_general, false);
         PreferenceManager.setDefaultValues(this, R.xml.pref_data_sync, false);
         PreferenceManager.setDefaultValues(this, R.xml.pref_notifications, false);
         PreferenceManager.setDefaultValues(this, R.xml.pref_data_source, false);
-        new IdempotentMigrations(getApplicationContext()).performAll();
+        new IdempotentMigrations(this).performAll();
+
+        //create index
+        SQLiteDatabase db = Cache.openDatabase();
+        db.execSQL("CREATE UNIQUE INDEX IF NOT EXISTS bgtimestampindex ON bgreadings(timestamp)");
     }
 }
