diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
index 4131d4a..dfbc869 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
@@ -381,8 +381,8 @@ private boolean overlapping(AlertType at, boolean allday, int startTime, int end
         return  st1 <= st2 && et1 > st2 ||
                 st1 <= st2 && (et2 < st2) && et2 > st1 || //2nd timeframe passes midnight
                 st2 <= st1 && et2 > st1 ||
-                st2 <= st1 && (et1 < st1) && et1 > st2; //1st timeframe passes midnight
-
+                st2 <= st1 && (et1 < st1) && et1 > st2 ||
+                (et1 < st1 && et2 < st2); //both timeframes pass midnight -> overlap at least at midnight
     }
 
     private double parseDouble(String str) {
