diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java b/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
index b88c99e..0b4c8c0 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
@@ -506,12 +506,10 @@ public static BgReading lastNoSenssor() {
                 .execute();
     }
 
-    public static List<BgReading> latestForGraph(int number, double startTime) {
-        DecimalFormat df = new DecimalFormat("#");
-        df.setMaximumFractionDigits(1);
+    public static List<BgReading> latestForGraph(int number, long startTime) {
         return new Select()
                 .from(BgReading.class)
-                .where("timestamp >= " + df.format(startTime))
+                .where("timestamp >= " + (startTime >= 0 ? startTime : 0))
                 .where("calculated_value != 0")
                 .where("raw_data != 0")
                 .orderBy("timestamp desc")
@@ -520,7 +518,7 @@ public static BgReading lastNoSenssor() {
     }
 
     public static List<BgReading> last30Minutes() {
-        double timestamp = (new Date().getTime()) - (60000 * 30);
+        long timestamp = (new Date().getTime()) - (60000 * 30);
         return new Select()
                 .from(BgReading.class)
                 .where("timestamp >= " + timestamp)
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java b/app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java
index 5dfdc21..34d61c0 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java
@@ -24,10 +24,8 @@
 import com.google.gson.internal.bind.DateTypeAdapter;
 
 import java.text.DecimalFormat;
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.List;
-import java.util.UUID;
+import java.text.NumberFormat;
+import java.util.*;
 
 /**
  * Created by stephenblack on 10/29/14.
@@ -637,12 +635,10 @@ public static double min_recent() {
                 .execute();
     }
 
-    public static List<Calibration> latestForGraph(int number, double startTime) {
-        DecimalFormat df = new DecimalFormat("#");
-        df.setMaximumFractionDigits(1);
+    public static List<Calibration> latestForGraph(int number, long startTime) {
         return new Select()
                 .from(Calibration.class)
-                .where("timestamp >= " + df.format(startTime))
+                .where("timestamp >= " + ((startTime >= 0) ? startTime : 0))
                 .orderBy("timestamp desc")
                 .limit(number)
                 .execute();
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
index 2196890..97696e6 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
@@ -37,8 +37,8 @@
  */
 public class BgGraphBuilder {
     public static final int FUZZER = (1000 * 30 * 5);
-    public double  end_time = (new Date().getTime() + (60000 * 10)) / FUZZER;
-    public double  start_time = end_time - ((60000 * 60 * 24)) / FUZZER;
+    public long end_time = (new Date().getTime() + (60000 * 10)) / FUZZER;
+    public long start_time = end_time - ((60000 * 60 * 24)) / FUZZER;
     public Context context;
     public SharedPreferences prefs;
     public double highMark;
