diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexShareCollectionService.java b/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexShareCollectionService.java
index 5a1c7c2..13041b3 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexShareCollectionService.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexShareCollectionService.java
@@ -535,7 +535,7 @@ private void gattWritingStep() {
             Log.d(TAG, "Writing: " + writePackets.get(index) + " index: " + index);
             if(mSendDataCharacteristic != null && writePackets != null) {
                 mSendDataCharacteristic.setValue(writePackets.get(index));
-                if (mBluetoothGatt.writeCharacteristic(mSendDataCharacteristic)) {
+                if (mBluetoothGatt != null && mBluetoothGatt.writeCharacteristic(mSendDataCharacteristic)) {
                     Log.d(TAG, "Wrote Successfully");
                 }
             }
