diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java
index ddd0383..12536a8 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java
@@ -264,6 +264,10 @@ private void Vibrate(Context ctx, AlertType alert, String bgValue, Boolean overr
         }
         if (profile != ALERT_PROFILE_SILENT && alert.vibrate) {
             builder.setVibrate(Notifications.vibratePattern);
+        } else {
+            // In order to still show on all android wear watches, either a sound or a vibrate pattern
+            // seems to be needed. This pattern basically does not vibrate:
+            builder.setVibrate(new long[]{1, 0});
         }
         NotificationManager mNotifyMgr = (NotificationManager) ctx.getSystemService(Context.NOTIFICATION_SERVICE);
         mNotifyMgr.cancel(Notifications.exportAlertNotificationId);
