diff --git a/app/build.gradle b/app/build.gradle
index 98a2276..ab307a9 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -85,6 +85,7 @@ android {
         targetSdkVersion 23
         versionCode 2
         versionName generateVersionName()
+        multiDexEnabled true
     }
     buildTypes {
     }
@@ -105,7 +106,7 @@ dependencies {
     compile 'com.github.lecho:hellocharts-android:v1.5.5'
     compile 'com.google.android.gms:play-services-wearable:7.5.0'
     compile 'com.google.guava:guava:18.0'
-
+    compile 'com.android.support:multidex:1.0.1'
     compile('com.github.nightscout:android-uploader:CORE_FOR_XDRIP') {
         transitive = false;
     }
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java b/app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java
index 693eb89..d33b84a 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java
@@ -4,6 +4,7 @@
 import android.preference.PreferenceManager;
 import android.content.Context;
 import android.content.Intent;
+import android.support.multidex.MultiDex;
 
 import com.eveningoutpost.dexdrip.Services.MissedReadingService;
 
@@ -33,4 +34,11 @@ public void onCreate() {
         context.startService(new Intent(context, MissedReadingService.class));	
         new IdempotentMigrations(getApplicationContext()).performAll();
     }
+
+    @Override
+    public void attachBaseContext(Context base) {
+        MultiDex.install(base);
+        super.attachBaseContext(base);
+    }
+
 }
