diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationOverride.java b/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationOverride.java
index 467d850..ac53a60 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationOverride.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationOverride.java
@@ -50,11 +50,7 @@ public void onClick(View v) {
                     String string_value = value.getText().toString();
                     if (!TextUtils.isEmpty(string_value)){
                         double calValue = Double.parseDouble(string_value);
-
-                        Calibration last_calibration = Calibration.last();
-                        last_calibration.sensor_confidence = 0;
-                        last_calibration.slope_confidence = 0;
-                        last_calibration.save();
+                        Calibration.clearLastCalibration(getApplicationContext());
                         Calibration.create(calValue, getApplicationContext());
 
                          Intent tableIntent = new Intent(v.getContext(), Home.class);
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java b/app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java
index bf3301d..3756b48 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java
@@ -559,6 +559,18 @@ public static void clear_all_existing_calibrations() {
         }
 
     }
+    
+    public static void clearLastCalibration(Context context) {
+    
+        Calibration last_calibration = Calibration.last();
+        if(last_calibration == null) {
+            return;
+        }
+        last_calibration.sensor_confidence = 0;
+        last_calibration.slope_confidence = 0;
+        last_calibration.save();
+        CalibrationSendQueue.addToQueue(last_calibration, context);
+    }
 
     public String toS() {
         Gson gson = new GsonBuilder()
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java
index 2d93511..dcc01c9 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java
@@ -289,6 +289,7 @@ private void populateV1APISensorEntry(JSONArray array, Sensor record) throws Exc
             SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ", Locale.US);
             format.setTimeZone(TimeZone.getDefault());
             json.put("type", "sensor");
+            json.put("date", record.started_at);
             json.put("xDrip_started_at", record.started_at);
             json.put("xDrip_stopped_at", record.stopped_at);
             json.put("xDrip_latest_battery_level", record.latest_battery_level);
@@ -364,6 +365,7 @@ private boolean doMongoUpload(SharedPreferences prefs, List<BgReading> glucoseDa
                             // make db object
                             BasicDBObject testData = new BasicDBObject();
                             testData.put("type", "sensor");
+                            testData.put("date", sensor.started_at);
                             testData.put("xDrip_started_at", sensor.started_at);
                             testData.put("xDrip_stopped_at", sensor.stopped_at);
                             testData.put("xDrip_latest_battery_level", sensor.latest_battery_level);
