diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
index 94d1da7..87d8071 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
@@ -90,9 +90,9 @@ public LineChartData previewLineData() {
         LineChartData previewLineData = new LineChartData(lineData());
         previewLineData.setAxisYLeft(yAxis());
         previewLineData.setAxisXBottom(previewXAxis());
-        previewLineData.getLines().get(4).setPointRadius(2);
         previewLineData.getLines().get(5).setPointRadius(2);
         previewLineData.getLines().get(6).setPointRadius(2);
+        previewLineData.getLines().get(7).setPointRadius(2);
         return previewLineData;
     }
 
