diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/xDripWidget.java b/app/src/main/java/com/eveningoutpost/dexdrip/xDripWidget.java
index c5dcde5..5b1dfe0 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/xDripWidget.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/xDripWidget.java
@@ -1,5 +1,6 @@
 package com.eveningoutpost.dexdrip;
 
+import android.app.PendingIntent;
 import android.appwidget.AppWidgetManager;
 import android.appwidget.AppWidgetProvider;
 import android.content.Context;
@@ -31,6 +32,14 @@ public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] a
         final int N = appWidgetIds.length;
         for (int i = 0; i < N; i++) {
             updateAppWidget(context, appWidgetManager, appWidgetIds[i]);
+
+            //Add behaviour: open xDrip on click
+            Intent intent = new Intent(context, Home.class);
+            int appWidgetId = appWidgetIds[i];
+            PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent, 0);
+            RemoteViews views = new RemoteViews(context.getPackageName(), R.layout.x_drip_widget);
+            views.setOnClickPendingIntent(R.id.xDripwidget, pendingIntent);;
+            appWidgetManager.updateAppWidget(appWidgetId, views);
         }
     }
 
diff --git a/app/src/main/res/layout/x_drip_widget.xml b/app/src/main/res/layout/x_drip_widget.xml
index 01b73ea..60aca0c 100644
--- a/app/src/main/res/layout/x_drip_widget.xml
+++ b/app/src/main/res/layout/x_drip_widget.xml
@@ -1,4 +1,5 @@
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+             android:id="@+id/xDripwidget"
              android:layout_width="wrap_content"
              android:layout_height="wrap_content"
              android:background="#55212121">
@@ -61,4 +62,4 @@
                 android:id="@+id/readingAge"
                 android:layout_gravity="center_horizontal"/>
     </LinearLayout>
-</FrameLayout>
\ No newline at end of file
+</FrameLayout>
