diff --git a/wear/src/main/java/com/eveningoutpost/dexdrip/BIGChart.java b/wear/src/main/java/com/eveningoutpost/dexdrip/BIGChart.java
index 8093d2b..1397bca 100644
--- a/wear/src/main/java/com/eveningoutpost/dexdrip/BIGChart.java
+++ b/wear/src/main/java/com/eveningoutpost/dexdrip/BIGChart.java
@@ -266,7 +266,7 @@ private void showAgoRawBatt() {
     }
 
     public void setColor() {
-        if (sharedPrefs.getBoolean("dark", false)) {
+        if (sharedPrefs.getBoolean("dark", true)) {
             setColorDark();
         } else {
             setColorBright();
diff --git a/wear/src/main/java/com/eveningoutpost/dexdrip/BaseWatchFace.java b/wear/src/main/java/com/eveningoutpost/dexdrip/BaseWatchFace.java
index 7d88e7e..408668b 100644
--- a/wear/src/main/java/com/eveningoutpost/dexdrip/BaseWatchFace.java
+++ b/wear/src/main/java/com/eveningoutpost/dexdrip/BaseWatchFace.java
@@ -259,7 +259,7 @@ private void showAgoRawBatt() {
     }
 
     public void setColor() {
-        if (sharedPrefs.getBoolean("dark", false)) {
+        if (sharedPrefs.getBoolean("dark", true)) {
             setColorDark();
         } else {
             setColorBright();
diff --git a/wear/src/main/java/com/eveningoutpost/dexdrip/CircleWatchface.java b/wear/src/main/java/com/eveningoutpost/dexdrip/CircleWatchface.java
index 86fac19..0b29c9d 100644
--- a/wear/src/main/java/com/eveningoutpost/dexdrip/CircleWatchface.java
+++ b/wear/src/main/java/com/eveningoutpost/dexdrip/CircleWatchface.java
@@ -337,7 +337,7 @@ protected void onTimeChanged(WatchFaceTime oldTime, WatchFaceTime newTime) {
 
     // defining color for dark and bright
     public int getLowColor() {
-        if (sharedPrefs.getBoolean("dark", false)) {
+        if (sharedPrefs.getBoolean("dark", true)) {
             return Color.argb(255, 255, 120, 120);
         } else {
             return Color.argb(255, 255, 80, 80);
