diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/ForegroundServiceStarter.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/ForegroundServiceStarter.java
index 1c6066b..c6a37fd 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/ForegroundServiceStarter.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/ForegroundServiceStarter.java
@@ -8,6 +8,8 @@
 import android.preference.PreferenceManager;
 import com.eveningoutpost.dexdrip.Models.UserError.Log;
 
+import java.util.Date;
+
 /**
  * Created by stephenblack on 12/25/14.
  */
@@ -32,7 +34,7 @@ public void start() {
             mHandler.post(new Runnable() {
                 @Override
                 public void run() {
-                    mService.startForeground(new Notifications().ongoingNotificationId, new Notifications().createOngoingNotification(new BgGraphBuilder(mContext), mContext));
+                    mService.startForeground(new Notifications().ongoingNotificationId, new Notifications().createOngoingNotification(mContext));
                 }
             });
         }
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
index dfa18f1..8616a28 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
@@ -245,9 +245,8 @@ boolean trendingToAlertEnd(Context context, Boolean newAlert, AlertType Alert) {
 
     private void notificationSetter(Context context) {
         ReadPerfs(context);
-        BgGraphBuilder bgGraphBuilder = new BgGraphBuilder(context);
         if (bg_ongoing && (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN)) {
-            bgOngoingNotification(bgGraphBuilder);
+            bgOngoingNotification();
         }
         if (prefs.getLong("alerts_disabled_until", 0) > new Date().getTime()) {
             Log.d("NOTIFICATIONS", "Notifications are currently disabled!!");
@@ -409,8 +408,11 @@ private Notification createExtensionPage(long hours) {
     }
 
     @TargetApi(Build.VERSION_CODES.JELLY_BEAN)
-    public Notification createOngoingNotification(BgGraphBuilder bgGraphBuilder, Context context) {
+    public Notification createOngoingNotification(Context context) {
         mContext = context;
+        long end = System.currentTimeMillis() + (60000 * 5);
+        long start = end - (60000 * 60*3) -  (60000 * 10);
+        BgGraphBuilder bgGraphBuilder = new BgGraphBuilder(mContext, start, end);
         ReadPerfs(mContext);
         Intent intent = new Intent(mContext, Home.class);
         List<BgReading> lastReadings = BgReading.latest(2);
@@ -462,13 +464,13 @@ public Notification createOngoingNotification(BgGraphBuilder bgGraphBuilder, Con
         return b.build();
     }
 
-    private void bgOngoingNotification(final BgGraphBuilder bgGraphBuilder) {
+    private void bgOngoingNotification() {
         mHandler.post(new Runnable() {
             @Override
             public void run() {
                 NotificationManagerCompat
                         .from(mContext)
-                        .notify(ongoingNotificationId, createOngoingNotification(bgGraphBuilder, mContext));
+                        .notify(ongoingNotificationId, createOngoingNotification(mContext));
                 if (iconBitmap != null)
                     iconBitmap.recycle();
                 if (notifiationBitmap != null)
