diff --git a/wear/src/main/java/com/eveningoutpost/dexdrip/BIGChart.java b/wear/src/main/java/com/eveningoutpost/dexdrip/BIGChart.java
index 1397bca..7f780fe 100644
--- a/wear/src/main/java/com/eveningoutpost/dexdrip/BIGChart.java
+++ b/wear/src/main/java/com/eveningoutpost/dexdrip/BIGChart.java
@@ -438,11 +438,11 @@ protected void setColorBright() {
     public void missedReadingAlert() {
         int minutes_since   = (int) Math.floor(timeSince()/(1000*60));
         if(minutes_since >= 16 && ((minutes_since - 16) % 5) == 0) {
-            NotificationCompat.Builder notification = new NotificationCompat.Builder(getApplicationContext())
+            /*NotificationCompat.Builder notification = new NotificationCompat.Builder(getApplicationContext())
                     .setContentTitle("Missed BG Readings")
                     .setVibrate(vibratePattern);
-            NotificationManager mNotifyMgr = (NotificationManager) getApplicationContext().getSystemService(getApplicationContext().NOTIFICATION_SERVICE);
-            mNotifyMgr.notify(missed_readings_alert_id, notification.build());
+            NotificationManager mNotifyMgr = (hNotificationManager) getApplicationContext().getSystemService(getApplicationContext().NOTIFICATION_SERVICE);
+            mNotifyMgr.notify(missed_readings_alert_id, notification.build());*/
             ListenerService.requestData(this); // attempt to recover missing data
         }
     }
diff --git a/wear/src/main/java/com/eveningoutpost/dexdrip/BaseWatchFace.java b/wear/src/main/java/com/eveningoutpost/dexdrip/BaseWatchFace.java
index 408668b..6d904ef 100644
--- a/wear/src/main/java/com/eveningoutpost/dexdrip/BaseWatchFace.java
+++ b/wear/src/main/java/com/eveningoutpost/dexdrip/BaseWatchFace.java
@@ -287,11 +287,11 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
     public void missedReadingAlert() {
         int minutes_since   = (int) Math.floor(timeSince()/(1000*60));
         if(minutes_since >= 16 && ((minutes_since - 16) % 5) == 0) {
-            NotificationCompat.Builder notification = new NotificationCompat.Builder(getApplicationContext())
+            /*NotificationCompat.Builder notification = new NotificationCompat.Builder(getApplicationContext())
                         .setContentTitle("Missed BG Readings")
                         .setVibrate(vibratePattern);
             NotificationManager mNotifyMgr = (NotificationManager) getApplicationContext().getSystemService(getApplicationContext().NOTIFICATION_SERVICE);
-            mNotifyMgr.notify(missed_readings_alert_id, notification.build());
+            mNotifyMgr.notify(missed_readings_alert_id, notification.build());*/
             ListenerService.requestData(this); // attempt to recover missing data
         }
     }
