diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java b/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java
index 5ab070b..e4bf640 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java
@@ -4,8 +4,10 @@
 import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.preference.PreferenceManager;
+import android.text.format.DateFormat;
 import android.view.View;
 import android.widget.Button;
+import android.widget.CheckBox;
 import android.widget.DatePicker;
 import android.widget.TimePicker;
 import android.widget.Toast;
@@ -23,6 +25,9 @@
     private Button button;
     private DatePicker dp;
     private TimePicker tp;
+    private CheckBox linkPickers;
+    
+    private int last_hour;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -32,13 +37,45 @@ protected void onCreate(Bundle savedInstanceState) {
             button = (Button)findViewById(R.id.startNewSensor);
             dp = (DatePicker)findViewById(R.id.datePicker);
             tp = (TimePicker)findViewById(R.id.timePicker);
+            tp.setIs24HourView(DateFormat.is24HourFormat(this));
+            tp.setSaveFromParentEnabled(false);
+            tp.setSaveEnabled(true);
             addListenerOnButton();
+            
+            tp.setOnTimeChangedListener(new TimePicker.OnTimeChangedListener() {
+
+                public void onTimeChanged(TimePicker arg0, int arg1, int arg2) {
+                    Log.d("NEW SENSOR", "new time " + arg1  + " " + arg2);
+
+                    if(arg1 == 23 && last_hour == 0) {
+                        Log.d("NEW SENSOR", "decreading day");
+                        addDays(-1);
+
+                    }
+                    if (arg1 == 0 && last_hour == 23) {
+                        Log.d("NEW SENSOR", "increasing day");
+                        addDays(1);
+                    }
+                    last_hour = arg1;
+
+                }
+            });
+
+            last_hour = tp.getCurrentHour();
+            
         } else {
             Intent intent = new Intent(this, StopSensor.class);
             startActivity(intent);
             finish();
         }
     }
+    
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+        prefs.edit().putBoolean("start_sensor_link_pickers", linkPickers.isChecked()).apply();
+    }
 
     @Override
     public String getMenuName() {
@@ -48,6 +85,9 @@ public String getMenuName() {
     public void addListenerOnButton() {
 
         button = (Button)findViewById(R.id.startNewSensor);
+        linkPickers = (CheckBox)findViewById(R.id.startSensorLinkPickers);
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+        linkPickers.setChecked(prefs.getBoolean("start_sensor_link_pickers", false));
 
         button.setOnClickListener(new View.OnClickListener() {
           public void onClick(View v) {
@@ -61,8 +101,11 @@ public void onClick(View v) {
               Log.d("NEW SENSOR", "Sensor started at " + startTime);
 
               Toast.makeText(getApplicationContext(), "NEW SENSOR STARTED", Toast.LENGTH_LONG).show();
-              CollectionServiceStarter.newStart(getApplicationContext());
+              
               SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+              prefs.edit().putBoolean("start_sensor_link_pickers", linkPickers.isChecked()).apply();
+              
+              CollectionServiceStarter.newStart(getApplicationContext());
               Intent intent;
               if(prefs.getBoolean("store_sensor_location",true)) {
                   intent = new Intent(getApplicationContext(), NewSensorLocation.class);
@@ -77,4 +120,19 @@ public void onClick(View v) {
         });
 
     }
+    
+    void addDays(int numberOfDays) {
+        
+        if(!linkPickers.isChecked()) {
+            return;
+        }
+        
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(dp.getYear(), dp.getMonth(), dp.getDayOfMonth(),
+        tp.getCurrentHour(), tp.getCurrentMinute(), 0);
+        
+        calendar.add(Calendar.DAY_OF_YEAR, numberOfDays);
+        dp.updateDate(calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH));
+    }
+
 }
diff --git a/app/src/main/res/layout/activity_start_new_sensor.xml b/app/src/main/res/layout/activity_start_new_sensor.xml
index 960316b..9c4893c 100644
--- a/app/src/main/res/layout/activity_start_new_sensor.xml
+++ b/app/src/main/res/layout/activity_start_new_sensor.xml
@@ -17,14 +17,6 @@
                 android:layout_width="fill_parent"
                 android:layout_height="fill_parent"
                 android:gravity="center_horizontal">
-                <DatePicker
-                    android:layout_width="wrap_content"
-                    android:layout_height="0dp"
-                    android:id="@+id/datePicker"
-                    android:datePickerMode="spinner"
-                    android:calendarViewShown="false"
-                    android:layout_weight="0.30" />
-
                 <TimePicker
                     android:layout_width="wrap_content"
                     android:layout_height="0dp"
@@ -35,6 +27,14 @@
                     android:timePickerMode="spinner"
                     android:layout_weight="0.30"/>
 
+                <DatePicker
+                    android:layout_width="wrap_content"
+                    android:layout_height="0dp"
+                    android:id="@+id/datePicker"
+                    android:datePickerMode="spinner"
+                    android:calendarViewShown="false"
+                    android:layout_weight="0.30" />
+
                 <TextView
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
@@ -56,6 +56,13 @@
                     android:layout_alignParentBottom="true"
                     android:layout_centerHorizontal="true"
                     android:layout_marginTop="15dp" />
+                
+                <CheckBox
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="link time and date pickers"
+                    android:id="@+id/startSensorLinkPickers"
+                    android:checked="false" />
             </LinearLayout>
         </ScrollView>
 
