diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
index a8fd1ea..86ade16 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
@@ -37,7 +37,7 @@
  * Created by stephenblack on 11/15/14.
  */
 public class BgGraphBuilder {
-    public static final int FUZZER = (1000 * 30 * 5);
+    public static final float FUZZER = (1000 * 30 * 5);
     public long  end_time;
     public long  start_time;
     public Context context;
@@ -452,7 +452,7 @@ public OnValueSelectTooltipListener getOnValueSelectTooltipListener(){
         public synchronized void onValueSelected(int i, int i1, PointValue pointValue) {
             final java.text.DateFormat timeFormat = DateFormat.getTimeFormat(context);
             //Won't give the exact time of the reading but the time on the grid: close enough.
-            Long time = ((long)pointValue.getX())*FUZZER;
+            Long time = (long)(pointValue.getX()*FUZZER);
             if(tooltip!= null){
                 tooltip.cancel();
             }
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSparklineBuilder.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSparklineBuilder.java
index 138213a..55b406c 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSparklineBuilder.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSparklineBuilder.java
@@ -32,8 +32,8 @@
     private int height;
     private BgGraphBuilder bgGraphBuilder;
     private LineChartView chart;
-    private long end = new Date().getTime() / BgGraphBuilder.FUZZER;
-    private long start = end - (60000*180 / BgGraphBuilder.FUZZER); // 3h
+    private long end = new Date().getTime() / (long)BgGraphBuilder.FUZZER;
+    private long start = end - (60000*180 / (long)BgGraphBuilder.FUZZER); // 3h
     private boolean showLowLine = false;
     private boolean showHighLine = false;
     private boolean showAxes = false;
@@ -41,12 +41,12 @@
     private boolean useTinyDots = false;
 
     public BgSparklineBuilder setStart(long start) {
-        this.start = start / BgGraphBuilder.FUZZER;
+        this.start = start / (long)BgGraphBuilder.FUZZER;
         return this;
     }
 
     public BgSparklineBuilder setEnd(long end) {
-        this.end = end / BgGraphBuilder.FUZZER;
+        this.end = end / (long)BgGraphBuilder.FUZZER;
         return this;
     }
 
