diff --git a/app/src/main/assets/migrations/27.sql b/app/src/main/assets/migrations/27.sql
deleted file mode 100644
index 2692b18..0000000
--- a/app/src/main/assets/migrations/27.sql
+++ /dev/null
@@ -1,3 +0,0 @@
-ALTER TABLE TransmitterData ADD COLUMN filtered_data DOUBLE;
-ALTER TABLE BgReadings ADD COLUMN filtered_data DOUBLE;
-ALTER TABLE BgReadings ADD COLUMN selected_filtered_data BOOLEAN;
diff --git a/app/src/main/assets/migrations/32.sql b/app/src/main/assets/migrations/32.sql
index 2eaae1d..f19ee24 100644
--- a/app/src/main/assets/migrations/32.sql
+++ b/app/src/main/assets/migrations/32.sql
@@ -1 +1 @@
-ALTER TABLE BgReadings ADD COLUMN selected_filtered_data BOOLEAN;
+ALTER TABLE BgReadings ADD COLUMN filtered_data DOUBLE;
