diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Services/G5CollectionService.java b/app/src/main/java/com/eveningoutpost/dexdrip/Services/G5CollectionService.java
index 3634b75..e407fce 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Services/G5CollectionService.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Services/G5CollectionService.java
@@ -68,6 +68,9 @@
 import java.util.Date;
 import java.util.List;
 import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 
 import javax.crypto.BadPaddingException;
 import javax.crypto.Cipher;
@@ -107,6 +110,9 @@
     private int lastBattery = 216;
     private long lastRead = new Date().getTime() - (5 * 60 *1000);
 
+    private static final ScheduledExecutorService worker =
+            Executors.newSingleThreadScheduledExecutor();
+
     private AlarmManager alarm;// = (AlarmManager) getSystemService(ALARM_SERVICE);
 
     private ScanSettings settings;
@@ -137,13 +143,15 @@ public void onCreate() {
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
+
 //        PowerManager powerManager = (PowerManager) getApplicationContext().getSystemService(POWER_SERVICE);
-//        PowerManager.WakeLock wakeLock = powerManager.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "DexShareCollectionStart");
-//        wakeLock.acquire(40000);
+//        PowerManager.WakeLock wakeLock = powerManager.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, TAG);
+//        wakeLock.acquire(4 * 60 * 1000);
+
         Log.d(TAG, "onG5StartCommand");
         Log.d(TAG, "SDK: " + Build.VERSION.SDK_INT);
         prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
-            defaultTransmitter = new Transmitter(prefs.getString("dex_txid", "ABCDEF"));
+        defaultTransmitter = new Transmitter(prefs.getString("dex_txid", "ABCDEF"));
         setMissedBgTimer();
 
         mBluetoothManager = (BluetoothManager) getSystemService(Context.BLUETOOTH_SERVICE);
@@ -209,7 +217,8 @@ public void stopScan() {
             if (Build.VERSION.SDK_INT < 21) {
                 mBluetoothAdapter.stopLeScan(mLeScanCallback);
             } else {
-                //mLEScanner.stopScan(mScanCallback);
+                Log.d(TAG, "stopScan");
+                mLEScanner.stopScan(mScanCallback);
             }
         }
     }
@@ -222,6 +231,17 @@ public void startScan() {
             mLEScanner.startScan(filters, settings, mScanCallback);
         }
     }
+    
+    void scanAfterDelay() {
+        
+        Runnable task = new Runnable() {
+            public void run() {
+                startScan();
+            }
+        };
+        worker.schedule(task, 12, TimeUnit.SECONDS);
+        
+    }
 
     private ScanCallback mScanCallback;
 
@@ -289,10 +309,10 @@ public void run() {
             };
 
     private void connectToDevice(BluetoothDevice device) {
-        //if (mGatt == null) {
-        mGatt = device.connectGatt(getApplicationContext(), false, gattCallback);
-        stopScan();
-        //}
+        if (mGatt == null) {
+            mGatt = device.connectGatt(getApplicationContext(), false, gattCallback);
+            stopScan();
+        }
     }
 
     private final BluetoothGattCallback gattCallback = new BluetoothGattCallback() {
@@ -305,10 +325,13 @@ public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState
                     break;
                 case BluetoothProfile.STATE_DISCONNECTED:
                     android.util.Log.e("gattCallback", "STATE_DISCONNECTED");
-                    device = null;
+                    if (mGatt == null) {
+                        scanAfterDelay();
+                        return;
+                    }
                     mGatt.close();
                     mGatt = null;
-                    startScan();
+                    scanAfterDelay();
                     break;
                 default:
                     android.util.Log.e("gattCallback", "STATE_OTHER");
diff --git a/app/src/main/res/layout/activity_home.xml b/app/src/main/res/layout/activity_home.xml
index 772057a..5484f75 100644
--- a/app/src/main/res/layout/activity_home.xml
+++ b/app/src/main/res/layout/activity_home.xml
@@ -31,7 +31,8 @@
                     android:layout_alignParentStart="false"
                     android:layout_alignParentEnd="true"
                     android:layout_height="match_parent"
-                    android:layout_below="@+id/extraStatusLine"/>
+                    android:layout_below="@+id/extraStatusLine"
+                    android:layout_marginTop="50dp"/>
 
                 <TextView
                     android:layout_width="900dp"
