diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java b/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java
index 4286008..ce84e26 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java
@@ -24,6 +24,7 @@
 import com.eveningoutpost.dexdrip.utils.LocationHelper;
 
 import java.util.Calendar;
+import java.util.Date;
 
 
 public class StartNewSensor extends ActivityWithMenu {
@@ -128,6 +129,11 @@ private void sensorButtonClick() {
         calendar.set(dp.getYear(), dp.getMonth(), dp.getDayOfMonth(), tp.getCurrentHour(), tp.getCurrentMinute(), 0);
         long startTime = calendar.getTime().getTime();
 
+        if(new Date().getTime() + 15 * 60000 < startTime ) {
+            Toast.makeText(getApplicationContext(), "ERROR: SENSOR START TIME IN FUTURE", Toast.LENGTH_LONG).show();
+            return;
+        }
+
         Sensor.create(startTime);
         Log.d("NEW SENSOR", "Sensor started at " + startTime);
 
