diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/wearintegration/SendToDataLayerThread.java b/app/src/main/java/com/eveningoutpost/dexdrip/wearintegration/SendToDataLayerThread.java
index 4c94566..91a182e 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/wearintegration/SendToDataLayerThread.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/wearintegration/SendToDataLayerThread.java
@@ -12,11 +12,14 @@
 import com.google.android.gms.wearable.PutDataRequest;
 import com.google.android.gms.wearable.Wearable;
 
+import java.util.concurrent.TimeUnit;
+
 /**
  * Created by stephenblack on 12/26/14.
  */
 class SendToDataLayerThread extends AsyncTask<DataMap,Void,Void> {
     private GoogleApiClient googleApiClient;
+    private static final String TAG = "SendDataThread";
     String path;
 
     SendToDataLayerThread(String path, GoogleApiClient pGoogleApiClient) {
@@ -26,21 +29,24 @@
 
     @Override
     protected Void doInBackground(DataMap... params) {
-        NodeApi.GetConnectedNodesResult nodes = Wearable.NodeApi.getConnectedNodes(googleApiClient).await();
-        for (Node node : nodes.getNodes()) {
-            for (DataMap dataMap : params) {
-                PutDataMapRequest putDMR = PutDataMapRequest.create(path);
-                putDMR.getDataMap().putAll(dataMap);
-                PutDataRequest request = putDMR.asPutDataRequest();
-                DataApi.DataItemResult result = Wearable.DataApi.putDataItem(googleApiClient,request).await();
-                if (result.getStatus().isSuccess()) {
-                    Log.d("SendDataThread", "DataMap: " + dataMap + " sent to: " + node.getDisplayName());
-                } else {
-                    Log.d("SendDataThread", "ERROR: failed to send DataMap");
+        try {
+            final NodeApi.GetConnectedNodesResult nodes = Wearable.NodeApi.getConnectedNodes(googleApiClient).await(15, TimeUnit.SECONDS);
+            for (Node node : nodes.getNodes()) {
+                for (DataMap dataMap : params) {
+                    PutDataMapRequest putDMR = PutDataMapRequest.create(path);
+                    putDMR.getDataMap().putAll(dataMap);
+                    PutDataRequest request = putDMR.asPutDataRequest();
+                    DataApi.DataItemResult result = Wearable.DataApi.putDataItem(googleApiClient, request).await(15, TimeUnit.SECONDS);
+                    if (result.getStatus().isSuccess()) {
+                        Log.d(TAG, "DataMap: " + dataMap + " sent to: " + node.getDisplayName());
+                    } else {
+                        Log.d(TAG, "ERROR: failed to send DataMap");
+                    }
                 }
             }
+        } catch (Exception e) {
+            Log.e(TAG, "Got exception sending data to wear: " + e.toString());
         }
-
         return null;
     }
 }
