diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java b/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
index f4ea467..ae6b1b2 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
@@ -383,7 +383,7 @@ static void updateCalculatedValue(BgReading bgReading ) {
 
     // Used by xDripViewer
     public static void create(Context context, double raw_data, double age_adjusted_raw_value, double filtered_data, Long timestamp,
-            double calculated_bg,  double calculated_current_slope, boolean hide_slope) {
+            double calculated_bg,  double calculated_current_slope, boolean hide_slope, double xdrip_filtered_calculated_value) {
         
         BgReading bgReading = new BgReading();
         Sensor sensor = Sensor.currentSensor();
@@ -411,6 +411,7 @@ public static void create(Context context, double raw_data, double age_adjusted_
         bgReading.calculated_value = calculated_bg;
         bgReading.calculated_value_slope = calculated_current_slope;
         bgReading.hide_slope = hide_slope;
+        bgReading.filtered_calculated_value = xdrip_filtered_calculated_value;
         bgReading.save();
 
         BgSendQueue.handleNewBgReading(bgReading, "create", context);
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Services/NsRestApiReader.java b/app/src/main/java/com/eveningoutpost/dexdrip/Services/NsRestApiReader.java
index 1007c97..36d8625 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Services/NsRestApiReader.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Services/NsRestApiReader.java
@@ -32,6 +32,7 @@
 	    double xDrip_calculated_value;
 	    double xDrip_calculated_current_slope;
 	    boolean xDrip_hide_slope;
+	    double xDrip_filtered_calculated_value;
 	    String device;
 	    String direction;
 	}
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Services/XDripViewer.java b/app/src/main/java/com/eveningoutpost/dexdrip/Services/XDripViewer.java
index c892234..7e130d8 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Services/XDripViewer.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Services/XDripViewer.java
@@ -165,7 +165,8 @@ private void readBgData(String baseUrl, String key) {
                     nightscoutBg.date, 
                     nightscoutBg.xDrip_calculated_value != 0 ? nightscoutBg.xDrip_calculated_value : nightscoutBg.sgv,
                     nightscoutBg.xDrip_calculated_current_slope,
-                    nightscoutBg.xDrip_hide_slope);
+                    nightscoutBg.xDrip_hide_slope,
+                    nightscoutBg.xDrip_filtered_calculated_value);
             
             lastInserted = nightscoutBg.date;
         }
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java
index e541088..15ba0b4 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java
@@ -218,6 +218,7 @@ private void populateV1APIBGEntry(JSONArray array, BgReading record, boolean xDr
             json.put("unfiltered", record.usedRaw() * 1000);
             json.put("rssi", 100);
             json.put("noise", record.noiseValue());
+            json.put("xDrip_filtered_calculated_value", record.filtered_calculated_value);
             if(xDripViewerMode) {
 	            json.put("xDrip_raw", record.raw_data);
 	            json.put("xDrip_filtered", record.filtered_data);
@@ -349,6 +350,7 @@ private boolean doMongoUpload(SharedPreferences prefs, List<BgReading> glucoseDa
                             testData.put("unfiltered", record.usedRaw() * 1000);
                             testData.put("rssi", 100);
                             testData.put("noise", record.noiseValue());
+                            testData.put("xDrip_filtered_calculated_value", record.filtered_calculated_value);
                             if(xDripViewerMode) {
 	                            testData.put("xDrip_raw", record.raw_data);
 	                            testData.put("xDrip_filtered", record.filtered_data);
