diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/AlertList.java b/app/src/main/java/com/eveningoutpost/dexdrip/AlertList.java
index a7db28e..2831ef5 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/AlertList.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/AlertList.java
@@ -1,6 +1,8 @@
 package com.eveningoutpost.dexdrip;
 
+import android.app.AlertDialog;
 import android.content.Context;
+import android.content.DialogInterface;
 import android.content.Intent;
 import android.content.SharedPreferences;
 import android.media.Ringtone;
@@ -168,9 +170,36 @@ public void onClick(View v) {
         });
     }
 
+    void displayWarning() {
+        runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+
+                if (!isFinishing()){
+                    new AlertDialog.Builder(AlertList.this)
+                      .setTitle("Warnining !")
+                      .setMessage("No active low alert exists, it is highly recomended that you will add a low an alert.")
+                      .setCancelable(false)
+                      .setPositiveButton(
+                              "Ok",
+                              new DialogInterface.OnClickListener() {
+                                  public void onClick(DialogInterface dialog, int id) {
+                                      dialog.cancel();
+                                  }
+                              })
+                    .create().show();
+                }
+            }
+        });
+ 
+    }
+
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        Log.d(TAG, "onActivityResult called ");
+        Log.d(TAG, "onActivityResult called request code  = " + requestCode + " result code " + resultCode);
+        if(!AlertType.activeLowAlertExists()) {
+            displayWarning();
+        }
         if (requestCode == ADD_ALERT || requestCode == EDIT_ALERT) {
             if(resultCode == RESULT_OK) {
                 Log.d(TAG, "onActivityResult called invalidating...");
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
index 6d20f6f..6133150 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
@@ -17,7 +17,6 @@
 import android.text.InputType;
 import android.text.format.DateFormat;
 import android.text.method.DigitsKeyListener;
-import com.eveningoutpost.dexdrip.Models.UserError.Log;
 import android.util.TypedValue;
 import android.view.MotionEvent;
 import android.view.View;
@@ -34,6 +33,7 @@
 import android.graphics.Paint;
 
 import com.eveningoutpost.dexdrip.Models.AlertType;
+import com.eveningoutpost.dexdrip.Models.UserError.Log;
 import com.eveningoutpost.dexdrip.UtilityModels.AlertPlayer;
 import com.eveningoutpost.dexdrip.UtilityModels.BgGraphBuilder;
 import com.eveningoutpost.dexdrip.UtilityModels.Constants;
@@ -88,7 +88,8 @@
     private String uuid;
     private Context mContext;
     private boolean above;
-    private final int CHOOSE_FILE = 1;
+    private final int REQUEST_CODE_CHOOSE_FILE = 1;
+    
     private final int MIN_ALERT = 40;
     private final int MAX_ALERT = 400;
 
@@ -112,7 +113,7 @@ String getExtra(Bundle savedInstanceState, String paramName, String defaultVal)
         	return defaultVal;
         }
     }
-
+    
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
@@ -456,7 +457,7 @@ private Integer parseInt(String str) {
     }
 
     public void addListenerOnButtons() {
-
+      
         buttonSave.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
                 // Check that values are ok.
@@ -561,7 +562,7 @@ public void onClick(DialogInterface dialog, int which) {
                                     Intent fileIntent = new Intent();
                                     fileIntent.setType("audio/mpeg3");
                                     fileIntent.setAction(Intent.ACTION_GET_CONTENT);
-                                    startActivityForResult(Intent.createChooser(fileIntent, "Select File for Alert"), CHOOSE_FILE);
+                                    startActivityForResult(Intent.createChooser(fileIntent, "Select File for Alert"), REQUEST_CODE_CHOOSE_FILE);
                                 } else {
                                     // Xdrip default was chossen, we live the file name as empty.
                                     audioPath = "";
@@ -649,7 +650,7 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                 audioPath = uri.toString();
                 alertMp3File.setText(shortPath(audioPath));
             } else {
-                if (requestCode == CHOOSE_FILE) {
+                if (requestCode == REQUEST_CODE_CHOOSE_FILE) {
                     Uri selectedImageUri = data.getData();
 
                     // Todo this code is very flacky. Probably need a much better understanding of how the different programs
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Models/AlertType.java b/app/src/main/java/com/eveningoutpost/dexdrip/Models/AlertType.java
index ce1dd69..7033672 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Models/AlertType.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Models/AlertType.java
@@ -363,7 +363,18 @@ public static void print_all() {
         return alerts;
     }
 
-
+    public static boolean activeLowAlertExists() {
+        List<AlertType> alerts = getAll(false);
+        if(alerts == null) {
+            return false;
+        }
+        for (AlertType alert : alerts) {
+            if(alert.active) {
+                return true;
+            }
+        }
+        return false;
+    }
 
     // This function is used to make sure that we always have a static alert on 55 low.
     // This alert will not be editable/removable.
