diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/AddCalibration.java b/app/src/main/java/com/eveningoutpost/dexdrip/AddCalibration.java
index f83c9d4..96674e4 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/AddCalibration.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/AddCalibration.java
@@ -16,7 +16,7 @@
 
 public class AddCalibration extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
     Button button;
-    private String menu_name = "Add Calibration";
+    public static String menu_name = "Add Calibration";
     private NavigationDrawerFragment mNavigationDrawerFragment;
 
     @Override
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/AlertList.java b/app/src/main/java/com/eveningoutpost/dexdrip/AlertList.java
index 2cb1025..d7c0aa2 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/AlertList.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/AlertList.java
@@ -14,6 +14,8 @@
 import android.view.Menu;
 import android.view.MenuItem;
 import android.view.View;
+import android.view.animation.Animation;
+import android.view.animation.AnimationUtils;
 import android.widget.AdapterView;
 import android.widget.Button;
 import android.widget.EditText;
@@ -26,6 +28,7 @@
 import com.eveningoutpost.dexdrip.Models.ActiveBgAlert;
 import com.eveningoutpost.dexdrip.Models.AlertType;
 import com.eveningoutpost.dexdrip.UtilityModels.AlertPlayer;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
 import java.text.DateFormat;
 import java.text.ParseException;
@@ -36,7 +39,8 @@
 import java.util.List;
 
 
-public class AlertList extends Activity {
+public class AlertList extends ActivityWithMenu {
+    public static String menu_name = "BG Level Alerts";
     ListView listViewLow;
     ListView listViewHigh;
     Button createLowAlert;
@@ -46,7 +50,7 @@
     final int ADD_ALERT = 1;
     final int EDIT_ALERT = 2;
     SharedPreferences prefs;
-
+    Animation anim;
     private final static String TAG = AlertPlayer.class.getSimpleName();
 
     String stringTimeFromAlert(AlertType alert) {
@@ -86,20 +90,36 @@ String stringTimeFromAlert(AlertType alert) {
 
 
     class AlertsOnItemLongClickListener implements AdapterView.OnItemLongClickListener {
-        //      @Override
-        public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
-            ListView lv = (ListView)parent;
-            @SuppressWarnings("unchecked")
-            HashMap<String, String> item = (HashMap<String, String>)lv.getItemAtPosition(position);
-            Log.e(TAG, "Item clicked " + lv.getItemAtPosition(position) + item.get("uuid"));
-
-            //The XML for each item in the list (should you use a custom XML) must have android:longClickable="true"
-            // as well (or you can use the convenience method lv.setLongClickable(true);). This way you can have a list
-            // with only some items responding to longclick. (might be used for non removable alerts)
-
-            Intent myIntent = new Intent(AlertList.this, EditAlertActivity.class);
-            myIntent.putExtra("uuid", item.get("uuid")); //Optional parameters
-            AlertList.this.startActivityForResult(myIntent, EDIT_ALERT);
+        @Override
+        public boolean onItemLongClick(final AdapterView<?> parent, final View view, final int position, final long id) {
+            anim.setAnimationListener(new Animation.AnimationListener() {
+
+                @Override
+                public void onAnimationStart(Animation animation) {
+                    ListView lv = (ListView) parent;
+                    @SuppressWarnings("unchecked")
+                    HashMap<String, String> item = (HashMap<String, String>) lv.getItemAtPosition(position);
+                    Log.e(TAG, "Item clicked " + lv.getItemAtPosition(position) + item.get("uuid"));
+
+                    //The XML for each item in the list (should you use a custom XML) must have android:longClickable="true"
+                    // as well (or you can use the convenience method lv.setLongClickable(true);). This way you can have a list
+                    // with only some items responding to longclick. (might be used for non removable alerts)
+
+                    Intent myIntent = new Intent(AlertList.this, EditAlertActivity.class);
+                    myIntent.putExtra("uuid", item.get("uuid")); //Optional parameters
+                    AlertList.this.startActivityForResult(myIntent, EDIT_ALERT);
+                }
+
+                @Override
+                public void onAnimationRepeat(Animation animation) {
+                }
+
+                @Override
+                public void onAnimationEnd(Animation animation) {
+
+                }
+            });
+            view.startAnimation(anim);
             return true;
         }
     }
@@ -116,11 +136,16 @@ protected void onCreate(Bundle savedInstanceState) {
 
         addListenerOnButton();
         FillLists();
-
+        anim = AnimationUtils.loadAnimation(this, R.anim.fade_anim);
         listViewLow.setOnItemLongClickListener(new AlertsOnItemLongClickListener());
         listViewHigh.setOnItemLongClickListener(new AlertsOnItemLongClickListener());
     }
 
+    @Override
+    public String getMenuName() {
+        return menu_name;
+    }
+
 
     public void addListenerOnButton() {
         createLowAlert = (Button)findViewById(R.id.button_create_low);
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java b/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
index b03014f..a6f5d97 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/BluetoothScan.java
@@ -25,13 +25,14 @@
 import com.eveningoutpost.dexdrip.Models.ActiveBluetoothDevice;
 import com.eveningoutpost.dexdrip.Services.DexCollectionService;
 import com.eveningoutpost.dexdrip.UtilityModels.CollectionServiceStarter;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
+import com.eveningoutpost.dexdrip.utils.ListActivityWithMenu;
 
 import java.util.ArrayList;
 
 @TargetApi(android.os.Build.VERSION_CODES.JELLY_BEAN_MR2)
-public class BluetoothScan extends ListActivity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
-    private String menu_name = "Scan for BT";
-    private NavigationDrawerFragment mNavigationDrawerFragment;
+public class BluetoothScan extends ListActivityWithMenu {
+    public static String menu_name = "Scan for BT";
 
     private final static String TAG = BluetoothScan.class.getSimpleName();
     private static final long SCAN_PERIOD = 10000;
@@ -89,15 +90,10 @@ protected void onPause() {
     }
 
     @Override
-    protected void onResume(){
-        super.onResume();
-        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
-        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
-    }
-    @Override
-    public void onNavigationDrawerItemSelected(int position) {
-        mNavigationDrawerFragment.swapContext(position);
+    public String getMenuName() {
+        return menu_name;
     }
+
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
         getMenuInflater().inflate(R.menu.menu_bluetooth_scan, menu);
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationCheckInActivity.java b/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationCheckInActivity.java
index 2a405b2..08738be 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationCheckInActivity.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationCheckInActivity.java
@@ -17,11 +17,11 @@
 import com.eveningoutpost.dexdrip.ImportedLibraries.dexcom.SyncingService;
 import com.eveningoutpost.dexdrip.R;
 import com.eveningoutpost.dexdrip.Sensor;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
-public class CalibrationCheckInActivity extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
-    private String menu_name = "Check in calibration";
-    private NavigationDrawerFragment mNavigationDrawerFragment;
-    Button button;
+public class CalibrationCheckInActivity extends ActivityWithMenu {
+    public static String menu_name = "Check in calibration";
+   Button button;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -29,15 +29,10 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_calibration_check_in);
         addListenerOnButton();
     }
-    protected void onResume(){
-        super.onResume();
-        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
-        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
-    }
 
     @Override
-    public void onNavigationDrawerItemSelected(int position) {
-        mNavigationDrawerFragment.swapContext(position);
+    public String getMenuName() {
+        return menu_name;
     }
 
     public void addListenerOnButton() {
@@ -54,7 +49,7 @@ public void onClick(View v) {
                     startActivity(tableIntent);
                     finish();
                 } else {
-                    Log.w("CANNOT CALIBRATE WITHOUT CURRENT SENSOR", "ERROR");
+                    Log.w("CALIBRATION", "ERROR, sensor not active");
                 }
             }
         });
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationGraph.java b/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationGraph.java
index 451034f..4cad3da 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationGraph.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationGraph.java
@@ -10,6 +10,7 @@
 
 import com.eveningoutpost.dexdrip.Models.Calibration;
 import com.eveningoutpost.dexdrip.UtilityModels.Constants;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
 import java.text.DateFormat;
 import java.text.DecimalFormat;
@@ -25,39 +26,28 @@
 import lecho.lib.hellocharts.view.LineChartView;
 
 
-public class CalibrationGraph extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
-    private String menu_name = "Calibration Graph";
-    private NavigationDrawerFragment mNavigationDrawerFragment;
-    private LineChartView chart;
+public class CalibrationGraph extends ActivityWithMenu {
+    public static String menu_name = "Calibration Graph";
+   private LineChartView chart;
     private LineChartData data;
     public double  start_x = 50;
     public double  end_x = 300;
-    
+
     TextView GraphHeader;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_calibration_graph);
-        
-        GraphHeader = (TextView) findViewById(R.id.CalibrationGraphHeader);
-    }
-    @Override
-    protected void onResume(){
-        super.onResume();
 
-        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
-        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
-
-        setupCharts();
+        GraphHeader = (TextView) findViewById(R.id.CalibrationGraphHeader);
     }
 
     @Override
-    public void onNavigationDrawerItemSelected(int position) {
-        mNavigationDrawerFragment.swapContext(position);
+    public String getMenuName() {
+        return menu_name;
     }
 
-
     public void setupCharts() {
         chart = (LineChartView) findViewById(R.id.chart);
         List<Calibration> calibrations = Calibration.allForSensor();
@@ -81,7 +71,7 @@ public void setupCharts() {
         if(calibration != null) {
             lineValues.add(new PointValue((float) start_x, (float) (start_x * calibration.slope + calibration.intercept)));
             lineValues.add(new PointValue((float) end_x, (float) (end_x * calibration.slope + calibration.intercept)));
-            
+
             DecimalFormat df = new DecimalFormat("#");
             df.setMaximumFractionDigits(2);
             df.setMinimumFractionDigits(2);
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationOverride.java b/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationOverride.java
index 1477233..79841b7 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationOverride.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/CalibrationOverride.java
@@ -15,12 +15,12 @@
 
 import com.eveningoutpost.dexdrip.Models.Calibration;
 import com.eveningoutpost.dexdrip.UtilityModels.CollectionServiceStarter;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
 
-public class CalibrationOverride extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
+public class CalibrationOverride extends ActivityWithMenu {
         Button button;
-    private String menu_name = "Override Calibration";
-    private NavigationDrawerFragment mNavigationDrawerFragment;
+    public static String menu_name = "Override Calibration";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -35,15 +35,8 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    protected void onResume(){
-                super.onResume();
-        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
-        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
-            }
-
-    @Override
-    public void onNavigationDrawerItemSelected(int position) {
-        mNavigationDrawerFragment.swapContext(position);
+    public String getMenuName() {
+        return menu_name;
     }
 
     public void addListenerOnButton() {
@@ -70,7 +63,7 @@ public void onClick(View v) {
                         value.setError("Calibration Can Not be blank");
                     }
                 } else {
-                    Log.w("CANNOT CALIBRATE WITHOUT CURRENT SENSOR", "ERROR");
+                    Log.w("Calibration", "ERROR, no active sensor");
                 }
             }
         });
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/DoubleCalibrationActivity.java b/app/src/main/java/com/eveningoutpost/dexdrip/DoubleCalibrationActivity.java
index 5c33b48..c96d8b3 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/DoubleCalibrationActivity.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/DoubleCalibrationActivity.java
@@ -15,12 +15,12 @@
 
 import com.eveningoutpost.dexdrip.Models.Calibration;
 import com.eveningoutpost.dexdrip.UtilityModels.CollectionServiceStarter;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
 
-public class DoubleCalibrationActivity  extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
+public class DoubleCalibrationActivity  extends ActivityWithMenu {
     Button button;
-    private String menu_name = "Add Double Calibration";
-    private NavigationDrawerFragment mNavigationDrawerFragment;
+    public static String menu_name = "Add Double Calibration";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -35,15 +35,8 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    protected void onResume(){
-        super.onResume();
-        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
-        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
-    }
-
-    @Override
-    public void onNavigationDrawerItemSelected(int position) {
-        mNavigationDrawerFragment.swapContext(position);
+    public String getMenuName() {
+        return menu_name;
     }
 
     public void addListenerOnButton() {
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
index 90478a9..76b6f6b 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java
@@ -48,8 +48,10 @@
 import com.eveningoutpost.dexdrip.UtilityModels.BgGraphBuilder;
 import com.eveningoutpost.dexdrip.UtilityModels.Constants;
 import com.eveningoutpost.dexdrip.R;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
-public class EditAlertActivity extends Activity {
+public class EditAlertActivity extends ActivityWithMenu {
+    public static String menu_name = "Edit Alert";
 
     TextView viewHeader;
 
@@ -245,6 +247,10 @@ protected void onCreate(Bundle savedInstanceState) {
 
     }
 
+    @Override
+    public String getMenuName() {
+        return menu_name;
+    }
     public static String UnitsConvert2Disp(boolean doMgdl, double threshold) {
         DecimalFormat df = new DecimalFormat("#");
         if(doMgdl ) {
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/FakeNumbers.java b/app/src/main/java/com/eveningoutpost/dexdrip/FakeNumbers.java
index 6e106af..d56eb1c 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/FakeNumbers.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/FakeNumbers.java
@@ -16,10 +16,13 @@
 import com.eveningoutpost.dexdrip.Models.ActiveBgAlert;
 import com.eveningoutpost.dexdrip.Models.AlertType;
 import com.eveningoutpost.dexdrip.Models.BgReading;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
 import java.util.Date;
 
-public class FakeNumbers extends Activity {
+public class FakeNumbers extends ActivityWithMenu {
+    public static String menu_name = "Fake Numbers";
+
     public Button button;
     public DatePicker dp;
     public TimePicker tp;
@@ -32,7 +35,10 @@ protected void onCreate(Bundle savedInstanceState) {
         addListenerOnButton();
 
     }
-
+    @Override
+    public String getMenuName() {
+        return menu_name;
+    }
     public void addListenerOnButton() {
 
         button = (Button)findViewById(R.id.log);
@@ -53,24 +59,24 @@ public void onClick(View v) {
             }
 
         });
-        
+
         button = (Button)findViewById(R.id.StartTest);
         button.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
                 ActiveBgAlert aba = ActiveBgAlert.getOnly();
                 ActiveBgAlert.ClearData();
                 ActiveBgAlert.Create("some string", true, new Date().getTime());
-                
-                
+
+
             }
         });
-        
+
         button = (Button)findViewById(R.id.StartTestAlerts);
         button.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
                 AlertType.testAll(getApplicationContext());
                 BgReading.TestgetUnclearTimes();
-                
+
             }
         });
     }
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Home.java b/app/src/main/java/com/eveningoutpost/dexdrip/Home.java
index ce7a4cf..c760661 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Home.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Home.java
@@ -35,6 +35,7 @@
 import com.eveningoutpost.dexdrip.UtilityModels.IdempotentMigrations;
 import com.eveningoutpost.dexdrip.UtilityModels.Intents;
 import com.eveningoutpost.dexdrip.UtilityModels.Notifications;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 import com.eveningoutpost.dexdrip.utils.DatabaseUtil;
 
 import java.io.File;
@@ -53,8 +54,8 @@
 import lecho.lib.hellocharts.view.PreviewLineChartView;
 
 
-public class Home extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
-    private String menu_name = "xDrip";
+public class Home extends ActivityWithMenu {
+    public static String menu_name = "xDrip";
     private NavigationDrawerFragment mNavigationDrawerFragment;
     private LineChartView chart;
     private PreviewLineChartView previewChart;
@@ -92,6 +93,11 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_home);
     }
 
+    @Override
+    public String getMenuName() {
+        return menu_name;
+    }
+
     public void checkEula() {
         boolean IUnderstand = prefs.getBoolean("I_understand", false);
         if (!IUnderstand) {
@@ -123,8 +129,6 @@ public void onReceive(Context ctx, Intent intent) {
         };
         registerReceiver(_broadcastReceiver, new IntentFilter(Intent.ACTION_TIME_TICK));
         registerReceiver(newDataReceiver, new IntentFilter(Intents.ACTION_NEW_BG_ESTIMATE_NO_DATA));
-        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
-        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
         holdViewport.set(0, 0, 0, 0);
         setupCharts();
         updateCurrentBgInfo();
@@ -179,11 +183,6 @@ public void onViewportChanged(Viewport newViewport) {
 
     }
 
-    @Override
-    public void onNavigationDrawerItemSelected(int position) {
-        mNavigationDrawerFragment.swapContext(position);
-    }
-
     public void setViewport() {
         if (tempViewport.left == 0.0 || holdViewport.left == 0.0 || holdViewport.right  >= (new Date().getTime())) {
             previewChart.setCurrentViewport(bgGraphBuilder.advanceViewport(chart, previewChart), false);
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java b/app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java
index 39b9a72..924f09a 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java
@@ -22,38 +22,35 @@
  * Created by stephenblack on 11/5/14.
  */
 public class NavDrawerBuilder {
-    public final List<Calibration> last_two_calibrations = Calibration.latest(2);
-    public final List<BgReading> last_two_bgReadings = BgReading.latestUnCalculated(2);
-    public final List<BgReading> bGreadings_in_last_30_mins = BgReading.last30Minutes();
-    public final boolean is_active_sensor = Sensor.isActive();
-    public final double time_now = new Date().getTime();
-    public List<Intent> nav_drawer_intents;
-    public List<String> nav_drawer_options;
+    public List<Calibration> last_two_calibrations = Calibration.latest(2);
+    public List<BgReading> last_two_bgReadings = BgReading.latestUnCalculated(2);
+    public List<BgReading> bGreadings_in_last_30_mins = BgReading.last30Minutes();
+    public boolean is_active_sensor = Sensor.isActive();
+    public double time_now = new Date().getTime();
+    public List<Intent> nav_drawer_intents = new ArrayList<>();
+    public List<String> nav_drawer_options = new ArrayList<>();
     public Context context;
 
     public NavDrawerBuilder(Context aContext) {
         context = aContext;
-        this.nav_drawer_intents = nav_drawer_intents();
-        this.nav_drawer_options = nav_drawer_options();
-
-    }
-
-    private final List<String> nav_drawer_options() {
-        List<String> options = new ArrayList<String>();
         SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
         boolean IUnderstand = prefs.getBoolean("I_understand", false);
         if(IUnderstand == false) {
-            options.add("Settings");
-            return options;
+            this.nav_drawer_options.add("Settings");
+            this.nav_drawer_intents.add(new Intent(context, Preferences.class));
+            return;
         }
 
-        options.add("xDrip");
+        this.nav_drawer_options.add(Home.menu_name);
+        this.nav_drawer_intents.add(new Intent(context, Home.class));
         if(is_active_sensor) {
-            options.add("Calibration Graph");
+            this.nav_drawer_options.add("Calibration Graph");
+            this.nav_drawer_intents.add(new Intent(context, CalibrationGraph.class));
         }
-        options.add("BG Data Table");
-        options.add("Calibration Data Table");
-//        options.add("Sensor Data Table");
+        this.nav_drawer_options.add("BG Data Table");
+        this.nav_drawer_intents.add(new Intent(context, BgReadingTable.class));
+        this.nav_drawer_options.add("Calibration Data Table");
+        this.nav_drawer_intents.add(new Intent(context, CalibrationDataTable.class));
 
         if(is_active_sensor) {
             if(!CollectionServiceStarter.isBTShare(context)) {
@@ -61,102 +58,50 @@ public NavDrawerBuilder(Context aContext) {
                     if (last_two_calibrations.size() > 1) {
                         if (bGreadings_in_last_30_mins.size() >= 2) {
                             if (time_now - last_two_calibrations.get(0).timestamp < (1000 * 60 * 60)) { //Put steps in place to discourage over calibration
-                                options.add("Override Calibration");
+                                this.nav_drawer_options.add(CalibrationOverride.menu_name);
+                                this.nav_drawer_intents.add(new Intent(context, CalibrationOverride.class));
                             } else {
-                                options.add("Add Calibration");
+                                this.nav_drawer_options.add(AddCalibration.menu_name);
+                                this.nav_drawer_intents.add(new Intent(context, AddCalibration.class));
                             }
                         } else {
-                            options.add("Cannot Calibrate right now");
-                        }
-                        if (last_two_calibrations.get(0).slope >= 1.4 || last_two_calibrations.get(0).slope <= 0.5) {
-                            options.add("Add Double Calibration");
+                            this.nav_drawer_options.add("Cannot Calibrate right now");
+                            this.nav_drawer_intents.add(new Intent(context, Home.class));
                         }
                     } else {
-                        options.add("Add Double Calibration");
+                        this.nav_drawer_options.add(DoubleCalibrationActivity.menu_name);
+                        this.nav_drawer_intents.add(new Intent(context, DoubleCalibrationActivity.class));
                     }
                 }
             }
-            options.add("Stop Sensor");
-        } else { options.add("Start Sensor"); }
-        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR2) {
-            if(CollectionServiceStarter.isBTWixel(context) || CollectionServiceStarter.isDexbridgeWixel(context)|| CollectionServiceStarter.isBTShare(context)) {
-                options.add("Scan for BT");
-            }
-        }
-        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR2) {
-            options.add("System Status");
-        }
-        boolean bg_alerts = prefs.getBoolean("bg_alerts_from_main_menu", false);
-        if (bg_alerts) {
-            options.add("BG Level Alerts");
-        }
-        options.add("Snooze Alerts");
-        options.add("Settings");
-//        options.add("Fake Numbers");
-//        options.add("Add Double Calibration");
-//        options.add("Share Test");
-        return options;
-    }
-
-    private List<Intent> nav_drawer_intents() {
-        List<Intent> options = new ArrayList<Intent>();
-        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
-        boolean IUnderstand = prefs.getBoolean("I_understand", false);
-        if(IUnderstand == false) {
-            options.add(new Intent(context, Preferences.class));
-            return options;
-        }
-
-        options.add(new Intent(context, Home.class));
-        if(is_active_sensor) {
-            options.add(new Intent(context, CalibrationGraph.class));
+            this.nav_drawer_options.add(StopSensor.menu_name);
+            this.nav_drawer_intents.add(new Intent(context, StopSensor.class));
+        } else {
+            this.nav_drawer_options.add(StartNewSensor.menu_name);
+            this.nav_drawer_intents.add(new Intent(context, StartNewSensor.class));
         }
-        options.add(new Intent(context, BgReadingTable.class));
-        options.add(new Intent(context, CalibrationDataTable.class));
-//        options.add(new Intent(context, SensorDataTable.class));
 
-
-        if(is_active_sensor) {
-            if(!CollectionServiceStarter.isBTShare(context)) {
-                if (last_two_bgReadings.size() > 1) {
-                    if (last_two_calibrations.size() > 1) {
-                        if (bGreadings_in_last_30_mins.size() >= 2) {
-                            if (time_now - last_two_calibrations.get(0).timestamp < (1000 * 60 * 60)) { //Put steps in place to discourage over calibration
-                                options.add(new Intent(context, CalibrationOverride.class));
-                            } else {
-                                options.add(new Intent(context, AddCalibration.class));
-                            }
-                        } else {
-                            options.add(new Intent(context, Home.class));
-                        }
-                        if (last_two_calibrations.get(0).slope >= 1.4 || last_two_calibrations.get(0).slope <= 0.5) {
-                            options.add(new Intent(context, DoubleCalibrationActivity.class));
-                        }
-                    } else {
-                        options.add(new Intent(context, DoubleCalibrationActivity.class));
-                    }
-                }
-            }
-            options.add(new Intent(context, StopSensor.class));
-        } else { options.add(new Intent(context, StartNewSensor.class)); }
         if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR2) {
             if(CollectionServiceStarter.isBTWixel(context) || CollectionServiceStarter.isDexbridgeWixel(context)|| CollectionServiceStarter.isBTShare(context)) {
-                options.add(new Intent(context, BluetoothScan.class));
+                this.nav_drawer_options.add(BluetoothScan.menu_name);
+                this.nav_drawer_intents.add(new Intent(context, BluetoothScan.class));
             }
         }
+
         if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR2) {
-            options.add(new Intent(context, SystemStatus.class));
+            this.nav_drawer_options.add(SystemStatus.menu_name);
+            this.nav_drawer_intents.add(new Intent(context, SystemStatus.class));
         }
+
         boolean bg_alerts = prefs.getBoolean("bg_alerts_from_main_menu", false);
         if (bg_alerts) {
-            options.add(new Intent(context, AlertList.class));
+            this.nav_drawer_options.add(AlertList.menu_name);
+            this.nav_drawer_intents.add(new Intent(context, AlertList.class));
         }
-        options.add(new Intent(context, SnoozeActivity.class));
-        options.add(new Intent(context, Preferences.class));
-//        options.add(new Intent(context, FakeNumbers.class));
-//        options.add(new Intent(context, DoubleCalibrationActivity.class));
-//        options.add(new Intent(context, ShareTest.class));
-        return options;
-    }
+        this.nav_drawer_options.add(SnoozeActivity.menu_name);
+        this.nav_drawer_intents.add(new Intent(context, SnoozeActivity.class));
 
+        this.nav_drawer_options.add("Settings");
+        this.nav_drawer_intents.add(new Intent(context, Preferences.class));
+    }
 }
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java b/app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java
index 0ccf8eb..81a974b 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java
@@ -30,9 +30,12 @@
 import com.eveningoutpost.dexdrip.Models.AlertType;
 import com.eveningoutpost.dexdrip.UtilityModels.AlertPlayer;
 import com.eveningoutpost.dexdrip.UtilityModels.BgGraphBuilder;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
 
-public class SnoozeActivity extends Activity {
+public class SnoozeActivity extends ActivityWithMenu {
+    public static String menu_name = "Snooze Alert";
+
     TextView alertStatus;
     Button buttonSnooze;
     Button disableAlerts;
@@ -115,6 +118,10 @@ protected void onCreate(Bundle savedInstanceState) {
 
         displayStatus();
     }
+    @Override
+    public String getMenuName() {
+        return menu_name;
+    }
 
     public void addListenerOnButton() {
         buttonSnooze = (Button)findViewById(R.id.button_snooze);
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java b/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java
index 983ea0a..8cbd8a5 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/StartNewSensor.java
@@ -14,14 +14,14 @@
 import android.widget.Toast;
 
 import com.eveningoutpost.dexdrip.UtilityModels.CollectionServiceStarter;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
 import java.util.Calendar;
 import java.util.List;
 
 
-public class StartNewSensor extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
-    private String menu_name = "Start Sensor";
-    private NavigationDrawerFragment mNavigationDrawerFragment;
+public class StartNewSensor extends ActivityWithMenu {
+    public static String menu_name = "Start Sensor";
     public Button button;
     public DatePicker dp;
     public TimePicker tp;
@@ -31,40 +31,20 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         if(Sensor.isActive() == false) {
             setContentView(R.layout.activity_start_new_sensor);
-
             button = (Button)findViewById(R.id.startNewSensor);
             dp = (DatePicker)findViewById(R.id.datePicker);
             tp = (TimePicker)findViewById(R.id.timePicker);
             addListenerOnButton();
-
-
         } else {
             Intent intent = new Intent(this, StopSensor.class);
             startActivity(intent);
             finish();
         }
     }
-    @Override
-    protected void onResume(){
-        super.onResume();
-        NavigationDrawerFragment mNavigationDrawerFragment;
-        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
-        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
-    }
 
     @Override
-    public void onNavigationDrawerItemSelected(int position) {
-        NavDrawerBuilder navDrawerBuilder = new NavDrawerBuilder(getApplicationContext());
-        List<String> menu_option_list = navDrawerBuilder.nav_drawer_options;
-        int menu_position = menu_option_list.indexOf(menu_name);
-
-        if (position != menu_position) {
-            List<Intent> intent_list = navDrawerBuilder.nav_drawer_intents;
-            Intent[] intent_array = intent_list.toArray(new Intent[intent_list.size()]);
-            startActivity(intent_array[position]);
-            finish();
-        }
-
+    public String getMenuName() {
+        return menu_name;
     }
 
     public void addListenerOnButton() {
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/StopSensor.java b/app/src/main/java/com/eveningoutpost/dexdrip/StopSensor.java
index 4d239e5..457ded3 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/StopSensor.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/StopSensor.java
@@ -12,11 +12,11 @@
 import android.widget.Toast;
 import java.util.Date;
 import com.eveningoutpost.dexdrip.UtilityModels.AlertPlayer;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
-public class StopSensor extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
-    private String menu_name = "Stop Sensor";
-    private NavigationDrawerFragment mNavigationDrawerFragment;
-    public Button button;
+public class StopSensor extends ActivityWithMenu {
+    public static String menu_name = "Stop Sensor";
+   public Button button;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -31,16 +31,10 @@ protected void onCreate(Bundle savedInstanceState) {
             addListenerOnButton();
         }
     }
-    @Override
-    protected void onResume(){
-        super.onResume();
-        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
-        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
-    }
 
     @Override
-    public void onNavigationDrawerItemSelected(int position) {
-        mNavigationDrawerFragment.swapContext(position);
+    public String getMenuName() {
+        return menu_name;
     }
 
     public void addListenerOnButton() {
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/SystemStatus.java b/app/src/main/java/com/eveningoutpost/dexdrip/SystemStatus.java
index 5b8e6e5..2b6cc3b 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/SystemStatus.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/SystemStatus.java
@@ -26,13 +26,13 @@
 import com.eveningoutpost.dexdrip.Models.ActiveBluetoothDevice;
 import com.eveningoutpost.dexdrip.Models.Calibration;
 import com.eveningoutpost.dexdrip.UtilityModels.CollectionServiceStarter;
+import com.eveningoutpost.dexdrip.utils.ActivityWithMenu;
 
 import java.lang.reflect.Method;
 
 
-public class SystemStatus extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
-    private String menu_name = "System Status";
-    private NavigationDrawerFragment mNavigationDrawerFragment;
+public class SystemStatus extends ActivityWithMenu {
+    public static String menu_name = "System Status";
     private TextView version_name_view;
     public TextView collection_method;
     public TextView current_device;
@@ -50,9 +50,7 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_system_status);
         prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
-        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
-        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
-        version_name_view = (TextView)findViewById(R.id.version_name);
+       version_name_view = (TextView)findViewById(R.id.version_name);
         collection_method = (TextView)findViewById(R.id.collection_method);
         connection_status = (TextView)findViewById(R.id.connection_status);
         current_device = (TextView)findViewById(R.id.remembered_device);
@@ -70,8 +68,8 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    public void onNavigationDrawerItemSelected(int position) {
-        mNavigationDrawerFragment.swapContext(position);
+    public String getMenuName() {
+        return menu_name;
     }
 
     private void set_current_values() {
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/utils/ActivityWithMenu.java b/app/src/main/java/com/eveningoutpost/dexdrip/utils/ActivityWithMenu.java
new file mode 100644
index 0000000..990531f
--- /dev/null
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/utils/ActivityWithMenu.java
@@ -0,0 +1,54 @@
+package com.eveningoutpost.dexdrip.utils;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.widget.DrawerLayout;
+
+import com.eveningoutpost.dexdrip.NavDrawerBuilder;
+import com.eveningoutpost.dexdrip.NavigationDrawerFragment;
+import com.eveningoutpost.dexdrip.R;
+
+import java.util.List;
+
+/**
+ * Created by stephenblack on 6/8/15.
+ */
+public class ActivityWithMenu extends Activity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
+    private NavDrawerBuilder navDrawerBuilder;
+    private List<Intent> intent_list;
+    private List<String> menu_option_list;
+    private int menu_position;
+    private String menu_name;
+    private NavigationDrawerFragment mNavigationDrawerFragment;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    protected void onResume(){
+        super.onResume();
+        menu_name = getMenuName();
+        navDrawerBuilder = new NavDrawerBuilder(getApplicationContext());
+        menu_option_list = navDrawerBuilder.nav_drawer_options;
+        intent_list = navDrawerBuilder.nav_drawer_intents;
+        menu_position = menu_option_list.indexOf(menu_name);
+
+        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
+        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
+    }
+
+    @Override
+    public void onNavigationDrawerItemSelected(int position) {
+        if (position != menu_position) {
+            startActivity(intent_list.get(position));
+            finish();
+        }
+    }
+
+    public String getMenuName() { //NOTE: THIS MUST BE OVERRIDEN!!!
+        throw new RuntimeException("MUST OVERRIDE THIS!");
+    }
+}
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/utils/ListActivityWithMenu.java b/app/src/main/java/com/eveningoutpost/dexdrip/utils/ListActivityWithMenu.java
new file mode 100644
index 0000000..e7f94fc
--- /dev/null
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/utils/ListActivityWithMenu.java
@@ -0,0 +1,54 @@
+package com.eveningoutpost.dexdrip.utils;
+
+import android.app.ListActivity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.widget.DrawerLayout;
+
+import com.eveningoutpost.dexdrip.NavDrawerBuilder;
+import com.eveningoutpost.dexdrip.NavigationDrawerFragment;
+import com.eveningoutpost.dexdrip.R;
+
+import java.util.List;
+
+/**
+ * Created by stephenblack on 6/8/15.
+ */
+public class ListActivityWithMenu extends ListActivity implements NavigationDrawerFragment.NavigationDrawerCallbacks {
+    private NavDrawerBuilder navDrawerBuilder;
+    private List<Intent> intent_list;
+    private List<String> menu_option_list;
+    private int menu_position;
+    private String menu_name;
+    private NavigationDrawerFragment mNavigationDrawerFragment;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    protected void onResume(){
+        super.onResume();
+        menu_name = getMenuName();
+        navDrawerBuilder = new NavDrawerBuilder(getApplicationContext());
+        menu_option_list = navDrawerBuilder.nav_drawer_options;
+        intent_list = navDrawerBuilder.nav_drawer_intents;
+        menu_position = menu_option_list.indexOf(menu_name);
+
+        mNavigationDrawerFragment = (NavigationDrawerFragment) getFragmentManager().findFragmentById(R.id.navigation_drawer);
+        mNavigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), menu_name, this);
+    }
+
+    @Override
+    public void onNavigationDrawerItemSelected(int position) {
+        if (position != menu_position) {
+            startActivity(intent_list.get(position));
+            finish();
+        }
+    }
+
+    public String getMenuName() { //NOTE: THIS MUST BE OVERRIDEN!!!
+        throw new RuntimeException("MUST OVERRIDE THIS!");
+    }
+}
diff --git a/app/src/main/res/anim/fade_anim.xml b/app/src/main/res/anim/fade_anim.xml
new file mode 100644
index 0000000..5f37504
--- /dev/null
+++ b/app/src/main/res/anim/fade_anim.xml
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<alpha xmlns:android="http://schemas.android.com/apk/res/android"
+    android:fromAlpha="1.0" android:toAlpha="0.0"
+    android:interpolator="@android:anim/linear_interpolator"
+    android:duration="300"/>
diff --git a/app/src/main/res/layout/activity_alert_list.xml b/app/src/main/res/layout/activity_alert_list.xml
index 31124f3..f228166 100644
--- a/app/src/main/res/layout/activity_alert_list.xml
+++ b/app/src/main/res/layout/activity_alert_list.xml
@@ -1,61 +1,71 @@
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    android:layout_width="fill_parent"
-    android:layout_height="fill_parent"
-    android:orientation="vertical"
-    android:gravity="center_horizontal"
-    android:showDividers="middle">
+<android.support.v4.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools" android:id="@+id/drawer_layout"
+    android:layout_width="match_parent" android:layout_height="match_parent">
 
-    <LinearLayout
-        android:orientation="vertical"
+    <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
         android:layout_width="fill_parent"
-        android:layout_weight=".46"
-        android:layout_height="0dp"
-        android:layout_gravity="center_horizontal"
+        android:layout_height="fill_parent"
+        android:orientation="vertical"
         android:gravity="center_horizontal"
-        android:weightSum="1">
-        <Button
-            android:id="@+id/button_create_low"
-            android:layout_width="fill_parent"
-            android:layout_height="wrap_content"
-            android:text="Create low alert" />
-        <ListView
-            android:id="@+id/listView_low"
-            android:layout_width="fill_parent"
-            android:layout_height="fill_parent"
-            android:layout_gravity="bottom"/>
-    </LinearLayout>
+        android:showDividers="middle">
 
-    <LinearLayout
-        android:orientation="vertical"
-        android:layout_width="fill_parent"
-        android:layout_weight=".46"
-        android:layout_height="0dp"
-        android:layout_gravity="center_horizontal"
-        android:gravity="center_horizontal">
-        <Button
-            android:id="@+id/button_create_high"
+        <LinearLayout
+            android:orientation="vertical"
             android:layout_width="fill_parent"
-            android:layout_height="wrap_content"
-            android:text="Create high alert"/>
-        <ListView
-            android:id="@+id/listView_high"
+            android:layout_weight=".46"
+            android:layout_height="0dp"
+            android:layout_gravity="center_horizontal"
+            android:gravity="center_horizontal"
+            android:weightSum="1">
+            <Button
+                android:id="@+id/button_create_low"
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:text="Create low alert" />
+            <ListView
+                android:id="@+id/listView_low"
+                android:layout_width="fill_parent"
+                android:layout_height="fill_parent"
+                android:layout_gravity="bottom"/>
+        </LinearLayout>
+
+        <LinearLayout
+            android:orientation="vertical"
             android:layout_width="fill_parent"
-            android:layout_height="fill_parent"/>
-    </LinearLayout>
+            android:layout_weight=".46"
+            android:layout_height="0dp"
+            android:layout_gravity="center_horizontal"
+            android:gravity="center_horizontal">
+            <Button
+                android:id="@+id/button_create_high"
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:text="Create high alert"/>
+            <ListView
+                android:id="@+id/listView_high"
+                android:layout_width="fill_parent"
+                android:layout_height="fill_parent"/>
+        </LinearLayout>
 
-    <LinearLayout
-        android:orientation="vertical"
-        android:layout_width="fill_parent"
-        android:layout_weight=".08"
-        android:layout_height="0dp"
-        android:gravity="center_horizontal"
-        android:showDividers="middle">
-        <TextView
+        <LinearLayout
+            android:orientation="vertical"
             android:layout_width="fill_parent"
-            android:layout_height="wrap_content"
-            android:text="(Long Press an existing alert to edit)"
-            android:textSize="12sp"
-            android:gravity="center_horizontal" />
-    </LinearLayout>
+            android:layout_weight=".08"
+            android:layout_height="0dp"
+            android:gravity="center_horizontal"
+            android:showDividers="middle">
+            <TextView
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:text="(Long Press an existing alert to edit)"
+                android:textSize="12sp"
+                android:gravity="center_horizontal" />
+        </LinearLayout>
 
-</LinearLayout>
+    </LinearLayout>
+<fragment android:id="@+id/navigation_drawer"
+android:layout_width="@dimen/navigation_drawer_width" android:layout_height="match_parent"
+android:layout_gravity="start"
+android:name="com.eveningoutpost.dexdrip.NavigationDrawerFragment"
+tools:layout="@layout/fragment_navigation_drawer" />
+    </android.support.v4.widget.DrawerLayout>
diff --git a/app/src/main/res/layout/activity_edit_alert.xml b/app/src/main/res/layout/activity_edit_alert.xml
index bb29de7..2ee7763 100644
--- a/app/src/main/res/layout/activity_edit_alert.xml
+++ b/app/src/main/res/layout/activity_edit_alert.xml
@@ -1,15 +1,12 @@
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools" android:layout_width="match_parent"
-    android:layout_height="match_parent" android:paddingLeft="@dimen/activity_horizontal_margin"
-    android:paddingRight="@dimen/activity_horizontal_margin"
-    android:paddingTop="@dimen/activity_vertical_margin"
-    android:paddingBottom="@dimen/activity_vertical_margin"
-    tools:context="com.eveningoutpost.dexdrip.FakeNumbers">
+<android.support.v4.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools" android:id="@+id/drawer_layout"
+    android:layout_width="match_parent" android:layout_height="match_parent">
+
     <LinearLayout
-        android:orientation="vertical"
-        android:layout_width="fill_parent"
-        android:layout_height="fill_parent"
-        android:gravity="center_horizontal">
+            android:orientation="vertical"
+            android:layout_width="fill_parent"
+            android:layout_height="fill_parent"
+            android:gravity="center_horizontal">
         <TextView
             android:id="@+id/view_alert_header"
             android:layout_width="fill_parent"
@@ -326,4 +323,11 @@
 
         </ScrollView>
     </LinearLayout>
-</RelativeLayout>
+
+    <fragment android:id="@+id/navigation_drawer"
+    android:layout_width="@dimen/navigation_drawer_width" android:layout_height="match_parent"
+    android:layout_gravity="start"
+    android:name="com.eveningoutpost.dexdrip.NavigationDrawerFragment"
+    tools:layout="@layout/fragment_navigation_drawer" />
+
+</android.support.v4.widget.DrawerLayout>
diff --git a/app/src/main/res/layout/activity_fake_numbers.xml b/app/src/main/res/layout/activity_fake_numbers.xml
index 521acd5..5d32844 100644
--- a/app/src/main/res/layout/activity_fake_numbers.xml
+++ b/app/src/main/res/layout/activity_fake_numbers.xml
@@ -1,61 +1,71 @@
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools" android:layout_width="match_parent"
-    android:layout_height="match_parent" android:paddingLeft="@dimen/activity_horizontal_margin"
-    android:paddingRight="@dimen/activity_horizontal_margin"
-    android:paddingTop="@dimen/activity_vertical_margin"
-    android:paddingBottom="@dimen/activity_vertical_margin"
-    tools:context="com.eveningoutpost.dexdrip.FakeNumbers">
-
-    <EditText
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:inputType="number"
-        android:ems="10"
-        android:id="@+id/bg_value"
-        android:autoText="false"
-        android:text=""
-        android:singleLine="true"
-        android:textAlignment="center"
-        android:textSize="24dp"
-        android:hint="Enter BG Value"
-        android:layout_above="@+id/save_calibration_button"
-        android:layout_alignParentStart="true" />
-
-    <Button
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="Done"
-        android:id="@+id/log"
-        android:layout_alignParentBottom="true"
-        android:layout_alignEnd="@+id/timePicker" />
-
-    <DatePicker
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:id="@+id/datePicker"
-        android:calendarViewShown="false"
-        android:layout_alignParentTop="true"
-        android:layout_centerHorizontal="true" />
-
-    <TimePicker
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:id="@+id/timePicker"
-        android:layout_centerVertical="true"
-        android:layout_alignParentStart="true" />
-    
+<android.support.v4.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools" android:id="@+id/drawer_layout"
+    android:layout_width="match_parent" android:layout_height="match_parent">
+    <RelativeLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent" android:paddingLeft="@dimen/activity_horizontal_margin"
+        android:paddingRight="@dimen/activity_horizontal_margin"
+        android:paddingTop="@dimen/activity_vertical_margin"
+        android:paddingBottom="@dimen/activity_vertical_margin"
+        tools:context="com.eveningoutpost.dexdrip.FakeNumbers">
+
+        <EditText
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:inputType="number"
+            android:ems="10"
+            android:id="@+id/bg_value"
+            android:autoText="false"
+            android:text=""
+            android:singleLine="true"
+            android:textAlignment="center"
+            android:textSize="24dp"
+            android:hint="Enter BG Value"
+            android:layout_above="@+id/save_calibration_button"
+            android:layout_alignParentStart="true" />
+
         <Button
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="StartTest"
-        android:id="@+id/StartTest"
-         android:layout_below="@+id/timePicker"/>
-        
-         <Button
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="StartTestAlerts"
-        android:id="@+id/StartTestAlerts"
-         android:layout_below="@+id/StartTest"/>
-
-</RelativeLayout>
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="Done"
+            android:id="@+id/log"
+            android:layout_alignParentBottom="true"
+            android:layout_alignEnd="@+id/timePicker" />
+
+        <DatePicker
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:id="@+id/datePicker"
+            android:calendarViewShown="false"
+            android:layout_alignParentTop="true"
+            android:layout_centerHorizontal="true" />
+
+        <TimePicker
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:id="@+id/timePicker"
+            android:layout_centerVertical="true"
+            android:layout_alignParentStart="true" />
+
+            <Button
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="StartTest"
+            android:id="@+id/StartTest"
+             android:layout_below="@+id/timePicker"/>
+
+             <Button
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="StartTestAlerts"
+            android:id="@+id/StartTestAlerts"
+             android:layout_below="@+id/StartTest"/>
+
+    </RelativeLayout>
+    <fragment android:id="@+id/navigation_drawer"
+        android:layout_width="@dimen/navigation_drawer_width" android:layout_height="match_parent"
+        android:layout_gravity="start"
+        android:name="com.eveningoutpost.dexdrip.NavigationDrawerFragment"
+        tools:layout="@layout/fragment_navigation_drawer" />
+
+</android.support.v4.widget.DrawerLayout>
diff --git a/app/src/main/res/layout/activity_home.xml b/app/src/main/res/layout/activity_home.xml
index db0deeb..b38023a 100644
--- a/app/src/main/res/layout/activity_home.xml
+++ b/app/src/main/res/layout/activity_home.xml
@@ -88,6 +88,5 @@
         android:layout_gravity="start"
         android:name="com.eveningoutpost.dexdrip.NavigationDrawerFragment"
         tools:layout="@layout/fragment_navigation_drawer" />
-
 </android.support.v4.widget.DrawerLayout>
 
diff --git a/app/src/main/res/layout/activity_snooze.xml b/app/src/main/res/layout/activity_snooze.xml
index 96145aa..18c2541 100644
--- a/app/src/main/res/layout/activity_snooze.xml
+++ b/app/src/main/res/layout/activity_snooze.xml
@@ -1,71 +1,81 @@
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    android:layout_width="fill_parent"
-    android:layout_height="fill_parent"
-    android:orientation="vertical"
-    android:gravity="center_horizontal"
-    android:showDividers="middle">
+<android.support.v4.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools" android:id="@+id/drawer_layout"
+    android:layout_width="match_parent" android:layout_height="match_parent">
     <LinearLayout
         android:layout_width="fill_parent"
-        android:layout_height="0dp"
-        android:layout_weight=".3"
-        android:orientation="vertical"
-        android:gravity="center_horizontal">
-        <Button
-            android:id="@+id/button_snooze"
-            android:layout_width="fill_parent"
-            android:layout_height="fill_parent"
-            android:text="@string/snooze"
-            android:layout_margin="20dp"
-            android:textSize="30sp" />
-    </LinearLayout>
-    <LinearLayout
-        android:layout_width="fill_parent"
-        android:layout_height="0dp"
-        android:layout_weight=".3"
-        android:orientation="vertical"
-        android:gravity="center_horizontal">
-
-        <NumberPicker
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
-            android:id="@+id/snooze"
-            android:layout_gravity="center_horizontal"
-            android:descendantFocusability="blocksDescendants"/>
-    </LinearLayout>
-    <LinearLayout
-        android:layout_width="fill_parent"
-        android:layout_height="0dp"
-        android:layout_weight=".3"
+        android:layout_height="fill_parent"
         android:orientation="vertical"
         android:gravity="center_horizontal"
-        android:paddingTop="10dp">
-
-        <TextView
-            android:id="@+id/alert_status"
+        android:showDividers="middle">
+        <LinearLayout
             android:layout_width="fill_parent"
-            android:layout_height="wrap_content"
-            android:gravity="center_horizontal"
-            android:text="replace dyanmicly (alert status)"
-            android:textSize="20sp"/>
+            android:layout_height="0dp"
+            android:layout_weight=".3"
+            android:orientation="vertical"
+            android:gravity="center_horizontal">
+            <Button
+                android:id="@+id/button_snooze"
+                android:layout_width="fill_parent"
+                android:layout_height="fill_parent"
+                android:text="@string/snooze"
+                android:layout_margin="20dp"
+                android:textSize="30sp" />
         </LinearLayout>
-    <LinearLayout
-        android:layout_width="fill_parent"
-        android:layout_height="0dp"
-        android:layout_weight=".1"
-        android:orientation="vertical"
-        android:gravity="center_horizontal">
-        <Button
-            android:id="@+id/button_disable_alerts"
+        <LinearLayout
             android:layout_width="fill_parent"
-            android:layout_height="wrap_content"
-            android:text="Disable All alerts"/>
-        <Button
-            android:id="@+id/enable_alerts"
+            android:layout_height="0dp"
+            android:layout_weight=".3"
+            android:orientation="vertical"
+            android:gravity="center_horizontal">
+
+            <NumberPicker
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:id="@+id/snooze"
+                android:layout_gravity="center_horizontal"
+                android:descendantFocusability="blocksDescendants"/>
+        </LinearLayout>
+        <LinearLayout
             android:layout_width="fill_parent"
-            android:layout_height="wrap_content"
-            android:text="Re-Enable Alerts"/>
-    </LinearLayout>
+            android:layout_height="0dp"
+            android:layout_weight=".3"
+            android:orientation="vertical"
+            android:gravity="center_horizontal"
+            android:paddingTop="10dp">
+
+            <TextView
+                android:id="@+id/alert_status"
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:gravity="center_horizontal"
+                android:text="replace dyanmicly (alert status)"
+                android:textSize="20sp"/>
+            </LinearLayout>
+        <LinearLayout
+            android:layout_width="fill_parent"
+            android:layout_height="0dp"
+            android:layout_weight=".1"
+            android:orientation="vertical"
+            android:gravity="center_horizontal">
+            <Button
+                android:id="@+id/button_disable_alerts"
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:text="Disable All alerts"/>
+            <Button
+                android:id="@+id/enable_alerts"
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:text="Re-Enable Alerts"/>
+        </LinearLayout>
+
 
 
+    </LinearLayout>
+    <fragment android:id="@+id/navigation_drawer"
+    android:layout_width="@dimen/navigation_drawer_width" android:layout_height="match_parent"
+    android:layout_gravity="start"
+    android:name="com.eveningoutpost.dexdrip.NavigationDrawerFragment"
+    tools:layout="@layout/fragment_navigation_drawer" />
 
-</LinearLayout>
+</android.support.v4.widget.DrawerLayout>
diff --git a/app/src/main/res/layout/activity_system_status.xml b/app/src/main/res/layout/activity_system_status.xml
index 3f4acc5..19a6322 100644
--- a/app/src/main/res/layout/activity_system_status.xml
+++ b/app/src/main/res/layout/activity_system_status.xml
@@ -18,161 +18,168 @@
         tools:context="com.eveningoutpost.dexdrip.SystemStatus"
         android:id="@+id/status_layout">
 
-        <LinearLayout
-            android:orientation="vertical"
+        <ScrollView
             android:layout_width="fill_parent"
             android:layout_height="fill_parent"
-            android:layout_alignParentTop="true"
-            android:layout_alignParentStart="true">
+            android:id="@+id/scrollView3" >
 
             <LinearLayout
-                android:orientation="horizontal"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:layout_gravity="center_horizontal"
-                android:paddingTop="10dp">
-
-                <TextView
-                    android:layout_width="wrap_content"
-                    android:layout_height="wrap_content"
-                    android:textAppearance="?android:attr/textAppearanceMedium"
-                    android:text="Version:   "
-                    android:id="@+id/textViewLabelVersionName" />
-
-                <TextView
-                    android:layout_width="wrap_content"
-                    android:layout_height="wrap_content"
-                    android:textAppearance="?android:attr/textAppearanceMedium"
-                    android:text="--"
-                    android:id="@+id/version_name" />
-            </LinearLayout>
-
-            <LinearLayout
-                android:orientation="horizontal"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:layout_gravity="center_horizontal"
-                android:id="@+id/layout_horiz"
-                android:paddingTop="10dp">
-
-                <TextView
-                    android:layout_width="wrap_content"
-                    android:layout_height="wrap_content"
-                    android:textAppearance="?android:attr/textAppearanceMedium"
-                    android:text="Collection Method:   "
-                    android:id="@+id/textView6" />
+                android:orientation="vertical"
+                android:layout_width="fill_parent"
+                android:layout_height="fill_parent"
+                android:layout_alignParentTop="true"
+                android:layout_alignParentStart="true">
 
-                <TextView
-                    android:layout_width="wrap_content"
+                <LinearLayout
+                    android:orientation="horizontal"
+                    android:layout_width="match_parent"
                     android:layout_height="wrap_content"
-                    android:textAppearance="?android:attr/textAppearanceMedium"
-                    android:id="@+id/collection_method" />
-            </LinearLayout>
+                    android:layout_gravity="center_horizontal"
+                    android:paddingTop="10dp">
 
-            <LinearLayout
-                android:orientation="horizontal"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:layout_gravity="center_horizontal"
-                android:paddingTop="10dp">
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:textAppearance="?android:attr/textAppearanceMedium"
+                        android:text="Version:   "
+                        android:id="@+id/textViewLabelVersionName" />
 
-                <TextView
-                    android:layout_width="wrap_content"
-                    android:layout_height="wrap_content"
-                    android:textAppearance="?android:attr/textAppearanceMedium"
-                    android:text="Remembered Device:   "
-                    android:id="@+id/textView7" />
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:textAppearance="?android:attr/textAppearanceMedium"
+                        android:text="--"
+                        android:id="@+id/version_name" />
+                </LinearLayout>
 
-                <TextView
-                    android:layout_width="wrap_content"
+                <LinearLayout
+                    android:orientation="horizontal"
+                    android:layout_width="match_parent"
                     android:layout_height="wrap_content"
-                    android:textAppearance="?android:attr/textAppearanceMedium"
-                    android:id="@+id/remembered_device" />
-            </LinearLayout>
+                    android:layout_gravity="center_horizontal"
+                    android:id="@+id/layout_horiz"
+                    android:paddingTop="10dp">
 
-            <LinearLayout
-                android:orientation="horizontal"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:paddingTop="10dp">
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:textAppearance="?android:attr/textAppearanceMedium"
+                        android:text="Collection Method:   "
+                        android:id="@+id/textView6" />
 
-                <TextView
-                    android:layout_width="wrap_content"
-                    android:layout_height="wrap_content"
-                    android:textAppearance="?android:attr/textAppearanceMedium"
-                    android:text="Connection Status:   "
-                    android:id="@+id/textView8" />
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:textAppearance="?android:attr/textAppearanceMedium"
+                        android:id="@+id/collection_method" />
+                </LinearLayout>
 
-                <TextView
-                    android:layout_width="wrap_content"
+                <LinearLayout
+                    android:orientation="horizontal"
+                    android:layout_width="match_parent"
                     android:layout_height="wrap_content"
-                    android:textAppearance="?android:attr/textAppearanceMedium"
-                    android:id="@+id/connection_status" />
-            </LinearLayout>
+                    android:layout_gravity="center_horizontal"
+                    android:paddingTop="10dp">
 
-            <TextView
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:textAppearance="?android:attr/textAppearanceLarge"
-                android:text="Other Notes:"
-                android:id="@+id/textView9"
-                android:layout_gravity="center_horizontal"
-                android:paddingTop="10dp" />
-
-            <TextView
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:textAppearance="?android:attr/textAppearanceMedium"
-                android:id="@+id/other_notes"
-                android:layout_gravity="center_horizontal" />
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:textAppearance="?android:attr/textAppearanceMedium"
+                        android:text="Remembered Device:   "
+                        android:id="@+id/textView7" />
 
-            <LinearLayout
-                android:orientation="vertical"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:layout_gravity="bottom"
-                android:gravity="center|bottom"
-                android:baselineAligned="false"
-                android:paddingTop="10dp">
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:textAppearance="?android:attr/textAppearanceMedium"
+                        android:id="@+id/remembered_device" />
+                </LinearLayout>
 
                 <LinearLayout
                     android:orientation="horizontal"
                     android:layout_width="match_parent"
                     android:layout_height="wrap_content"
-                    android:layout_gravity="center_horizontal|bottom"
-                    android:gravity="center_horizontal|bottom">
+                    android:paddingTop="10dp">
 
-                    <Button
-                        style="?android:attr/buttonStyleSmall"
+                    <TextView
                         android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
-                        android:text="Restart Collector"
-                        android:id="@+id/restart_collection_service" />
+                        android:textAppearance="?android:attr/textAppearanceMedium"
+                        android:text="Connection Status:   "
+                        android:id="@+id/textView8" />
 
-                    <Button
-                        style="?android:attr/buttonStyleSmall"
+                    <TextView
                         android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
-                        android:text="Forget Device"
-                        android:id="@+id/forget_device" />
+                        android:textAppearance="?android:attr/textAppearanceMedium"
+                        android:id="@+id/connection_status" />
                 </LinearLayout>
 
-                <LinearLayout
-                    android:orientation="horizontal"
+                <TextView
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
-                    android:gravity="center_horizontal"
-                    android:layout_gravity="center_horizontal">
+                    android:textAppearance="?android:attr/textAppearanceLarge"
+                    android:text="Other Notes:"
+                    android:id="@+id/textView9"
+                    android:layout_gravity="center_horizontal"
+                    android:paddingTop="10dp" />
 
-                    <ImageButton
+                <TextView
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:textAppearance="?android:attr/textAppearanceMedium"
+                    android:id="@+id/other_notes"
+                    android:layout_gravity="center_horizontal" />
+
+                <LinearLayout
+                    android:orientation="vertical"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="bottom"
+                    android:gravity="center|bottom"
+                    android:baselineAligned="false"
+                    android:paddingTop="10dp">
+
+                    <LinearLayout
+                        android:orientation="horizontal"
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="center_horizontal|bottom"
+                        android:gravity="center_horizontal|bottom">
+
+                        <Button
+                            style="?android:attr/buttonStyleSmall"
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:text="Restart Collector"
+                            android:id="@+id/restart_collection_service" />
+
+                        <Button
+                            style="?android:attr/buttonStyleSmall"
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:text="Forget Device"
+                            android:id="@+id/forget_device" />
+                    </LinearLayout>
+
+                    <LinearLayout
+                        android:orientation="horizontal"
                         android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
-                        android:id="@+id/refresh_current_values"
-                        android:src="@android:drawable/stat_notify_sync" />
+                        android:gravity="center_horizontal"
+                        android:layout_gravity="center_horizontal">
+
+                        <ImageButton
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:id="@+id/refresh_current_values"
+                            android:src="@android:drawable/stat_notify_sync" />
+                    </LinearLayout>
                 </LinearLayout>
+
             </LinearLayout>
+        </ScrollView>
 
-        </LinearLayout>
     </RelativeLayout>
 
     <fragment android:id="@+id/navigation_drawer"
diff --git a/app/src/main/res/layout/row_alerts.xml b/app/src/main/res/layout/row_alerts.xml
index c0e8b6b..e4f9560 100644
--- a/app/src/main/res/layout/row_alerts.xml
+++ b/app/src/main/res/layout/row_alerts.xml
@@ -2,7 +2,8 @@
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="fill_parent"
     android:layout_height="wrap_content"
-    android:orientation="vertical" >
+    android:orientation="vertical"
+    android:longClickable="true">
 
 
     <LinearLayout
