diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java
index f04463b..f71c1bc 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java
@@ -47,8 +47,8 @@
  */
 public class NightscoutUploader {
         private static final String TAG = NightscoutUploader.class.getSimpleName();
-        private static final int SOCKET_TIMEOUT = 60000;
-        private static final int CONNECTION_TIMEOUT = 30000;
+        private static final int SOCKET_TIMEOUT = 600000;
+        private static final int CONNECTION_TIMEOUT = 300000;
         private Context mContext;
         private Boolean enableRESTUpload;
         private Boolean enableMongoUpload;
