diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java b/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java
index 3538449..f438888 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java
@@ -222,6 +222,7 @@ public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState
             } else if (newState == BluetoothProfile.STATE_DISCONNECTED) {
                 mConnectionState = STATE_DISCONNECTED;
                 ActiveBluetoothDevice.disconnected();
+                lastdata = null;
                 Log.w(TAG, "onConnectionStateChange: Disconnected from GATT server.");
                 setRetryTimer();
             }
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
index 3438025..6ba5291 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java
@@ -335,6 +335,8 @@ public String unitizedDeltaString(double value) {
         if(doMgdl) {
             return delta_sign + df.format(unitized(value)) + " mg/dl";
         } else {
+            df.setMinimumFractionDigits(1);
+            df.setMinimumIntegerDigits(1);
             return delta_sign + df.format(unitized(value)) + " mmol";
         }
     }
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java
index e0e2a6b..1cbc2e6 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java
@@ -171,27 +171,16 @@ public int getBatteryLevel() {
     }
 
     public String slopeOrdinal(){
-        double slope_by_minute = mBgReading.calculated_value_slope * 60000;
-        String arrow = "0";
-        if (slope_by_minute <= (-3.5)) {
-            arrow = "7";
-        } else if (slope_by_minute <= (-2)) {
-            arrow = "6";
-        } else if (slope_by_minute <= (-1)) {
-            arrow = "5";
-        } else if (slope_by_minute <= (1)) {
-            arrow = "4";
-        } else if (slope_by_minute <= (2)) {
-            arrow = "3";
-        } else if (slope_by_minute <= (3.5)) {
-            arrow = "2";
-        } else {
-            arrow = "1";
-        }
-        if(mBgReading.hide_slope) {
-            arrow = "9";
-        }
-        return arrow;
+        String arrow_name = mBgReading.slopeName();
+        if(arrow_name.compareTo("DoubleDown")==0) return "7";
+        if(arrow_name.compareTo("SingleDown")==0) return "6";
+        if(arrow_name.compareTo("FortyFiveDown")==0) return "5";
+        if(arrow_name.compareTo("Flat")==0) return "4";
+        if(arrow_name.compareTo("FortyFiveUp")==0) return "3";
+        if(arrow_name.compareTo("SingleUp")==0) return "2";
+        if(arrow_name.compareTo("DoubleUp")==0) return "1";
+        if(arrow_name.compareTo("9")==0) return arrow_name;
+        return "0";
     }
 }
 
diff --git a/app/src/main/res/values/arrays.xml b/app/src/main/res/values/arrays.xml
index c6461dd..b450c67 100644
--- a/app/src/main/res/values/arrays.xml
+++ b/app/src/main/res/values/arrays.xml
@@ -2,7 +2,7 @@
 <resources>
     <string-array name="DexCollectionMethods">
         <item>Bluetooth Wixel</item>
-        <item>Dexbridge</item>
+        <item>xBridge Wixel</item>
         <item>Wifi Wixel</item>
         <item>DexcomShare</item>
     </string-array>
