diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java b/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java
index a61942c..8b559b2 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java
@@ -141,6 +141,11 @@ public void onSharedPreferenceChanged(SharedPreferences prefs, String key) {
                     Log.w(TAG, "Removing from foreground");
                 }
             }
+            if(key.equals("dex_collection_method") || key.equals("dex_txid")){
+                //if the input method or ID changed, accept any new package once even if they seem duplicates
+                Log.d(TAG, "collection method or txID changed - setting lastdata to null");
+                lastdata = null;
+            }
         }
     };
 
