diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java
index 1cbc2e6..58ef742 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java
@@ -86,9 +86,12 @@ public void receiveData(final Context context, final int transactionId, final Pe
                 Log.d(TAG, "receiveData: transactionId is " + String.valueOf(transactionId));
                 if (lastTransactionId == 0 || transactionId != lastTransactionId) {
                     lastTransactionId = transactionId;
-                    Log.d(TAG, "Received Query. data: " + data.size());
+                    Log.d(TAG, "Received Query. data: " + data.size() + ". sending ACK and data");
                     PebbleKit.sendAckToPebble(context, transactionId);
                     sendData();
+                } else {
+                    Log.d(TAG, "receiveData: lastTransactionId is "+ String.valueOf(lastTransactionId)+ ", sending NACK");
+                    PebbleKit.sendNackToPebble(context,transactionId);
                 }
             }
         });
