diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java
index a71c6ef..82f635b 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java
@@ -94,14 +94,6 @@ public static void addToQueue(BgReading bgReading, String operation_type, Contex
                 context.startService(new Intent(context, widgetUpdateService.class));
             }
 
-            if (prefs.getBoolean("cloud_storage_mongodb_enable", false) || prefs.getBoolean("cloud_storage_api_enable", false)) {
-                Log.w("SENSOR QUEUE:", String.valueOf(bgSendQueue.mongo_success));
-                if (operation_type.compareTo("create") == 0) {
-                    MongoSendTask task = new MongoSendTask(context, bgSendQueue);
-                    task.execute();
-                }
-            }
-
             if (prefs.getBoolean("broadcast_data_through_intents", false)) {
                 Log.i("SENSOR QUEUE:", "Broadcast data");
                 final Bundle bundle = new Bundle();
diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CalibrationSendQueue.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CalibrationSendQueue.java
index f355a40..7a94964 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CalibrationSendQueue.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CalibrationSendQueue.java
@@ -59,13 +59,6 @@ public static void addToQueue(Calibration calibration, Context context) {
         calibrationSendQueue.success = false;
         calibrationSendQueue.mongo_success = false;
         calibrationSendQueue.save();
-
-        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
-
-        if (prefs.getBoolean("cloud_storage_mongodb_enable", false) || prefs.getBoolean("cloud_storage_api_enable", false)) {
-            MongoSendTask task = new MongoSendTask(context, calibrationSendQueue);
-            task.execute();
-        }
     }
 
     public void markMongoSuccess() {
