diff --git a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
index a7ab1f9..f1e1fbb 100644
--- a/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
+++ b/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
@@ -222,6 +222,10 @@ boolean trendingToAlertEnd(Context context, AlertType Alert) {
     // only function that is really called from outside...
     public void notificationSetter(Context context) {
         ReadPerfs(context);
+        BgGraphBuilder bgGraphBuilder = new BgGraphBuilder(context);
+        if (bg_ongoing && (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN)) {
+            bgOngoingNotification(bgGraphBuilder);
+        }
         if(prefs.getLong("alerts_disabled_until", 0) > new Date().getTime()){
             Log.w("NOTIFICATIONS", "Notifications are currently disabled!!");
             return;
@@ -230,7 +234,6 @@ public void notificationSetter(Context context) {
         BgReading.checkForDropAllert(context);
         BgReading.checkForRisingAllert(context);
 
-        BgGraphBuilder bgGraphBuilder = new BgGraphBuilder(context);
         Sensor sensor = Sensor.currentSensor();
 
         List<BgReading> bgReadings = BgReading.latest(3);
@@ -238,9 +241,6 @@ public void notificationSetter(Context context) {
         if(bgReadings == null || bgReadings.size() < 3) { return; }
         if(calibrations == null || calibrations.size() < 2) { return; }
         BgReading bgReading = bgReadings.get(0);
-        if (bg_ongoing && (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN)) {
-            bgOngoingNotification(bgGraphBuilder);
-        }
 
         if (calibration_notifications) {
             if (bgReadings.size() >= 3) {
